{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "bf8a7126",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import seaborn as sns\n",
    "import Bio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eab066b0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "id": "9a48f9c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'C:\\\\Users\\\\MitoClubRoom\\\\Documents\\\\Eldar'"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.chdir('C:/Users/MitoClubRoom/Documents/Eldar')\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "8b7d0036",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VF_chr1_fasta:\n",
      "NC_006840.2\n",
      "Seq('AAGATCACTTAATATATATAAGATCTTTTAAAGAGATCTTTTATTAGATCTATT...AGA')\n",
      "Lenght of chromosome 1:  2897536\n",
      "number of A: 882812\n",
      "number of T: 886048\n",
      "number of G: 563483\n",
      "number of C: 565193\n",
      "VF_chr2_fasta:\n",
      "NC_006841.2\n",
      "Seq('ACTATGAAAACAATTGTACAGCATAAAAGCAATAATTCAACAAATAGCTGAACA...TTA')\n",
      "Lenght of chromosome 2:  1330333\n",
      "number of A: 423081\n",
      "number of T: 414726\n",
      "number of G: 245073\n",
      "number of C: 247453\n",
      "VC_chr1_fasta:\n",
      "NZ_CP016324.1\n",
      "Seq('GGCAAAATGTGTGAGGATCTTAGTTATCGGTCGAAAAATAATGTGAATAACTTA...ACG')\n",
      "Lenght of chromosome 1:  2987030\n",
      "number of A: 781763\n",
      "number of T: 776269\n",
      "number of G: 714454\n",
      "number of C: 714544\n",
      "VC_chr2_fasta:\n",
      "CP016325.1\n",
      "Seq('GATGAAAAACCGTCTATCCTAACAAGTCTCAGTCAAAACAAGATAAACAGAACA...AGT')\n",
      "Lenght of chromosome 2:  1101931\n",
      "number of A: 291986\n",
      "number of T: 293941\n",
      "number of G: 260201\n",
      "number of C: 255803\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "VF_chr1_fasta = SeqIO.read(\"VF_chr_1.fasta\", \"fasta\")\n",
    "print(\"VF_chr1_fasta:\")\n",
    "print(VF_chr1_fasta.id)\n",
    "print(repr(VF_chr1_fasta.seq))\n",
    "print('Lenght of chromosome 1: ', len(VF_chr1_fasta))\n",
    "VF_chr1_A_count = VF_chr1_fasta.count(\"A\")\n",
    "print('number of A:', VF_chr1_A_count)\n",
    "VF_chr1_T_count = VF_chr1_fasta.count(\"T\")\n",
    "print('number of T:', VF_chr1_T_count)\n",
    "VF_chr1_G_count = VF_chr1_fasta.count(\"G\")\n",
    "print('number of G:',VF_chr1_G_count)\n",
    "VF_chr1_C_count = VF_chr1_fasta.count(\"C\")\n",
    "print('number of C:', VF_chr1_C_count)\n",
    "VF_chr2_fasta = SeqIO.read(\"VF_chr_2.fasta\", \"fasta\")\n",
    "print(\"VF_chr2_fasta:\")\n",
    "print(VF_chr2_fasta.id)\n",
    "print(repr(VF_chr2_fasta.seq))\n",
    "print('Lenght of chromosome 2: ', len(VF_chr2_fasta))\n",
    "VF_chr2_A_count = VF_chr2_fasta.count(\"A\")\n",
    "print('number of A:', VF_chr2_A_count)\n",
    "VF_chr2_T_count = VF_chr2_fasta.count(\"T\")\n",
    "print('number of T:', VF_chr2_T_count)\n",
    "VF_chr2_G_count = VF_chr2_fasta.count(\"G\")\n",
    "print('number of G:', VF_chr2_G_count)\n",
    "VF_chr2_C_count = VF_chr2_fasta.count(\"C\")\n",
    "print('number of C:', VF_chr2_C_count)\n",
    "\n",
    "VC_chr1_fasta = SeqIO.read(\"VC_chr_1.fasta\", \"fasta\")\n",
    "print(\"VC_chr1_fasta:\")\n",
    "print(VC_chr1_fasta.id)\n",
    "print(repr(VC_chr1_fasta.seq))\n",
    "print('Lenght of chromosome 1: ', len(VC_chr1_fasta))\n",
    "VC_chr1_A_count = VC_chr1_fasta.count(\"A\")\n",
    "print('number of A:', VC_chr1_A_count)\n",
    "VC_chr1_T_count = VC_chr1_fasta.count(\"T\")\n",
    "print('number of T:', VC_chr1_T_count)\n",
    "VC_chr1_G_count = VC_chr1_fasta.count(\"G\")\n",
    "print('number of G:', VC_chr1_G_count)\n",
    "VC_chr1_C_count = VC_chr1_fasta.count(\"C\")\n",
    "print('number of C:', VC_chr1_C_count)\n",
    "\n",
    "VC_chr2_fasta = SeqIO.read(\"VC_chr_2.fasta\", \"fasta\")\n",
    "print(\"VC_chr2_fasta:\")\n",
    "print(VC_chr2_fasta.id)\n",
    "print(repr(VC_chr2_fasta.seq))\n",
    "print('Lenght of chromosome 2: ', len(VC_chr2_fasta))\n",
    "VC_chr2_A_count = VC_chr2_fasta.count(\"A\")\n",
    "print('number of A:', VC_chr2_A_count)\n",
    "VC_chr2_T_count = VC_chr2_fasta.count(\"T\")\n",
    "print('number of T:', VC_chr2_T_count)\n",
    "VC_chr2_G_count = VC_chr2_fasta.count(\"G\")\n",
    "print('number of G:', VC_chr2_G_count)\n",
    "VC_chr2_C_count = VC_chr2_fasta.count(\"C\")\n",
    "print('number of C:', VC_chr2_C_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "1352be64",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Seq('AAGATCACTTAATATATATAAGATCTTTTAAAGAGATCTTTTATTAGATCTATT...TTT')\n",
      "VF_chr1_left_p, A 438645\n",
      "VF_chr1_left_p, G 302214\n",
      "VF_chr1_left_p, T 448583\n",
      "VF_chr1_left_p, C 259326\n",
      "Seq('TCAGAATGATTTTTCACAGACAATACCTAATTATGAAATAAAACGGTTGATACC...AGA')\n",
      "VF_chr1_right_p, T 444167\n",
      "VF_chr1_right_p, C 261269\n",
      "VF_chr1_right_p, A 437465\n",
      "VF_chr1_right_p, G 305867\n",
      "VF_chr1_count_A 876110\n",
      "VF_chr1_count_G 608081\n",
      "VF_chr1_count_T 892750\n",
      "VF_chr1_count_C 520595\n"
     ]
    }
   ],
   "source": [
    "VF_chr1_left_p = VF_chr1_fasta[:1448768]\n",
    "print(repr(VF_chr1_left_p.seq))\n",
    "VF_chr1_left_p_A = VF_chr1_left_p.count(\"A\")\n",
    "print('VF_chr1_left_p, A', VF_chr1_left_p_A)\n",
    "VF_chr1_left_p_G = VF_chr1_left_p.count(\"G\")\n",
    "print('VF_chr1_left_p, G', VF_chr1_left_p_G)\n",
    "VF_chr1_left_p_T = VF_chr1_left_p.count(\"T\")\n",
    "print('VF_chr1_left_p, T', VF_chr1_left_p_T)\n",
    "VF_chr1_left_p_C = VF_chr1_left_p.count(\"C\")\n",
    "print('VF_chr1_left_p, C', VF_chr1_left_p_C)\n",
    "\n",
    "VF_chr1_right_p = VF_chr1_fasta[1448768:]\n",
    "print(repr(VF_chr1_right_p.seq))\n",
    "VF_chr1_right_p_T = VF_chr1_right_p.count(\"A\")\n",
    "print('VF_chr1_right_p, T', VF_chr1_right_p_T)\n",
    "VF_chr1_right_p_C = VF_chr1_right_p.count(\"G\")\n",
    "print('VF_chr1_right_p, C', VF_chr1_right_p_C)\n",
    "VF_chr1_right_p_A = VF_chr1_right_p.count(\"T\")\n",
    "print('VF_chr1_right_p, A', VF_chr1_right_p_A)\n",
    "VF_chr1_right_p_G = VF_chr1_right_p.count(\"C\")\n",
    "print('VF_chr1_right_p, G', VF_chr1_right_p_G)\n",
    "\n",
    "VF_chr1_count_A = VF_chr1_left_p_A + VF_chr1_right_p_A\n",
    "print(\"VF_chr1_count_A\", VF_chr1_count_A )\n",
    "VF_chr1_count_G = VF_chr1_left_p_G + VF_chr1_right_p_G\n",
    "print(\"VF_chr1_count_G\", VF_chr1_count_G )\n",
    "VF_chr1_count_T = VF_chr1_left_p_T + VF_chr1_right_p_T\n",
    "print(\"VF_chr1_count_T\", VF_chr1_count_T )\n",
    "VF_chr1_count_C = VF_chr1_left_p_C + VF_chr1_right_p_C\n",
    "print(\"VF_chr1_count_C\", VF_chr1_count_C )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "dd09af49",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Seq('ACTATGAAAACAATTGTACAGCATAAAAGCAATAATTCAACAAATAGCTGAACA...AAT')\n",
      "VF_chr2_left_p, A 208188\n",
      "VF_chr2_left_p, G 133398\n",
      "VF_chr2_left_p, T 209574\n",
      "VF_chr2_left_p, C 114006\n",
      "Seq('ACCTTAAATATTTTCATAATTTGATTATTTAAAGAGCGATTCTGTAGGAAGTAG...TTA')\n",
      "VF_chr2_right_p, T 214893\n",
      "VF_chr2_right_p, C 111675\n",
      "VF_chr2_right_p, A 205152\n",
      "VF_chr2_right_p, G 133447\n",
      "VF_chr2_count_A 413340\n",
      "VF_chr2_count_G 266845\n",
      "VF_chr2_count_T 424467\n",
      "VF_chr2_count_C 225681\n"
     ]
    }
   ],
   "source": [
    "VF_chr2_left_p = VF_chr2_fasta[:665166]\n",
    "print(repr(VF_chr2_left_p.seq))\n",
    "VF_chr2_left_p_A = VF_chr2_left_p.count(\"A\")\n",
    "print('VF_chr2_left_p, A', VF_chr2_left_p_A)\n",
    "VF_chr2_left_p_G = VF_chr2_left_p.count(\"G\")\n",
    "print('VF_chr2_left_p, G', VF_chr2_left_p_G)\n",
    "VF_chr2_left_p_T = VF_chr2_left_p.count(\"T\")\n",
    "print('VF_chr2_left_p, T', VF_chr2_left_p_T)\n",
    "VF_chr2_left_p_C = VF_chr2_left_p.count(\"C\")\n",
    "print('VF_chr2_left_p, C', VF_chr2_left_p_C)\n",
    "\n",
    "VF_chr2_right_p = VF_chr2_fasta[665166:]\n",
    "print(repr(VF_chr2_right_p.seq))\n",
    "VF_chr2_right_p_T = VF_chr2_right_p.count(\"A\")\n",
    "print('VF_chr2_right_p, T', VF_chr2_right_p_T)\n",
    "VF_chr2_right_p_C = VF_chr2_right_p.count(\"G\")\n",
    "print('VF_chr2_right_p, C', VF_chr2_right_p_C)\n",
    "VF_chr2_right_p_A = VF_chr2_right_p.count(\"T\")\n",
    "print('VF_chr2_right_p, A', VF_chr2_right_p_A)\n",
    "VF_chr2_right_p_G = VF_chr2_right_p.count(\"C\")\n",
    "print('VF_chr2_right_p, G', VF_chr2_right_p_G)\n",
    "\n",
    "VF_chr2_count_A = VF_chr2_left_p_A + VF_chr2_right_p_A\n",
    "print(\"VF_chr2_count_A\", VF_chr2_count_A)\n",
    "VF_chr2_count_G = VF_chr2_left_p_G + VF_chr2_right_p_G\n",
    "print(\"VF_chr2_count_G\", VF_chr2_count_G)\n",
    "VF_chr2_count_T = VF_chr2_left_p_T + VF_chr2_right_p_T\n",
    "print(\"VF_chr2_count_T\", VF_chr2_count_T)\n",
    "VF_chr2_count_C = VF_chr2_left_p_C + VF_chr2_right_p_C\n",
    "print(\"VF_chr2_count_C\", VF_chr2_count_C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "id": "b2ebaab0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Seq('GGCAAAATGTGTGAGGATCTTAGTTATCGGTCGAAAAATAATGTGAATAACTTA...CGA')\n",
      "VC_chr1_left_p, A 371449\n",
      "VC_chr1_left_p, G 367754\n",
      "VC_chr1_left_p, T 378575\n",
      "VC_chr1_left_p, C 330737\n",
      "Seq('TGGGGTTATGGAAGACGAATTTTACCCAAACCATCAAGCGGTCGATTTTTATCA...ACG')\n",
      "VC_chr1_right_p, T 410314\n",
      "VC_chr1_right_p, C 346700\n",
      "VC_chr1_right_p, A 397694\n",
      "VC_chr1_right_p, G 383807\n",
      "VC_chr1_count_A 769143\n",
      "VC_chr1_count_G 751561\n",
      "VC_chr1_count_T 788889\n",
      "VC_chr1_count_C 677437\n"
     ]
    }
   ],
   "source": [
    "VC_chr1_left_p = VC_chr1_fasta[:1448515]\n",
    "print(repr(VC_chr1_left_p.seq))\n",
    "VC_chr1_left_p_A = VC_chr1_left_p.count(\"A\")\n",
    "print('VC_chr1_left_p, A', VC_chr1_left_p_A)\n",
    "VC_chr1_left_p_G = VC_chr1_left_p.count(\"G\")\n",
    "print('VC_chr1_left_p, G', VC_chr1_left_p_G)\n",
    "VC_chr1_left_p_T = VC_chr1_left_p.count(\"T\")\n",
    "print('VC_chr1_left_p, T', VC_chr1_left_p_T)\n",
    "VC_chr1_left_p_C = VC_chr1_left_p.count(\"C\")\n",
    "print('VC_chr1_left_p, C', VC_chr1_left_p_C)\n",
    "\n",
    "VC_chr1_right_p = VC_chr1_fasta[1448515:]\n",
    "print(repr(VC_chr1_right_p.seq))\n",
    "VC_chr1_right_p_T = VC_chr1_right_p.count(\"A\")\n",
    "print('VC_chr1_right_p, T', VC_chr1_right_p_T)\n",
    "VC_chr1_right_p_C = VC_chr1_right_p.count(\"G\")\n",
    "print('VC_chr1_right_p, C', VC_chr1_right_p_C)\n",
    "VC_chr1_right_p_A = VC_chr1_right_p.count(\"T\")\n",
    "print('VC_chr1_right_p, A', VC_chr1_right_p_A)\n",
    "VC_chr1_right_p_G = VC_chr1_right_p.count(\"C\")\n",
    "print('VC_chr1_right_p, G', VC_chr1_right_p_G)\n",
    "\n",
    "VC_chr1_count_A = VC_chr1_left_p_A + VC_chr1_right_p_A\n",
    "print(\"VC_chr1_count_A\", VC_chr1_count_A )\n",
    "VC_chr1_count_G = VC_chr1_left_p_G + VC_chr1_right_p_G\n",
    "print(\"VC_chr1_count_G\", VC_chr1_count_G )\n",
    "VC_chr1_count_T = VC_chr1_left_p_T + VC_chr1_right_p_T\n",
    "print(\"VC_chr1_count_T\", VC_chr1_count_T )\n",
    "VC_chr1_count_C = VC_chr1_left_p_C + VC_chr1_right_p_C\n",
    "print(\"VC_chr1_count_C\", VC_chr1_count_C )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "id": "b341eb64",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Seq('GATGAAAAACCGTCTATCCTAACAAGTCTCAGTCAAAACAAGATAAACAGAACA...ATA')\n",
      "VC_chr2_left_p, A 147673\n",
      "VC_chr2_left_p, G 135582\n",
      "VC_chr2_left_p, T 153369\n",
      "VC_chr2_left_p, C 119341\n",
      "Seq('AACTGATCTCTTTCAACGCGGTAAATGAGCCAAACTGTTTTACCACATGACGAA...AGT')\n",
      "VC_chr2_right_p, T 144313\n",
      "VC_chr2_right_p, C 124619\n",
      "VC_chr2_right_p, A 140572\n",
      "VC_chr2_right_p, G 136462\n",
      "VC_chr2_count_A 288245\n",
      "VC_chr2_count_G 272044\n",
      "VC_chr2_count_T 297682\n",
      "VC_chr2_count_C 243960\n"
     ]
    }
   ],
   "source": [
    "VC_chr2_left_p = VC_chr2_fasta[:555965]\n",
    "print(repr(VC_chr2_left_p.seq))\n",
    "VC_chr2_left_p_A = VC_chr2_left_p.count(\"A\")\n",
    "print('VC_chr2_left_p, A', VC_chr2_left_p_A)\n",
    "VC_chr2_left_p_G = VC_chr2_left_p.count(\"G\")\n",
    "print('VC_chr2_left_p, G', VC_chr2_left_p_G)\n",
    "VC_chr2_left_p_T = VC_chr2_left_p.count(\"T\")\n",
    "print('VC_chr2_left_p, T', VC_chr2_left_p_T)\n",
    "VC_chr2_left_p_C = VC_chr2_left_p.count(\"C\")\n",
    "print('VC_chr2_left_p, C', VC_chr2_left_p_C)\n",
    "\n",
    "VC_chr2_right_p = VC_chr2_fasta[555965:]\n",
    "print(repr(VC_chr2_right_p.seq))\n",
    "VC_chr2_right_p_T = VC_chr2_right_p.count(\"A\")\n",
    "print('VC_chr2_right_p, T', VC_chr2_right_p_T)\n",
    "VC_chr2_right_p_C = VC_chr2_right_p.count(\"G\")\n",
    "print('VC_chr2_right_p, C', VC_chr2_right_p_C)\n",
    "VC_chr2_right_p_A = VC_chr2_right_p.count(\"T\")\n",
    "print('VC_chr2_right_p, A', VC_chr2_right_p_A)\n",
    "VC_chr2_right_p_G = VC_chr2_right_p.count(\"C\")\n",
    "print('VC_chr2_right_p, G', VC_chr2_right_p_G)\n",
    "\n",
    "VC_chr2_count_A = VC_chr2_left_p_A + VC_chr2_right_p_A\n",
    "print(\"VC_chr2_count_A\", VC_chr2_count_A)\n",
    "VC_chr2_count_G = VC_chr2_left_p_G + VC_chr2_right_p_G\n",
    "print(\"VC_chr2_count_G\", VC_chr2_count_G)\n",
    "VC_chr2_count_T = VC_chr2_left_p_T + VC_chr2_right_p_T\n",
    "print(\"VC_chr2_count_T\", VC_chr2_count_T)\n",
    "VC_chr2_count_C = VC_chr2_left_p_C + VC_chr2_right_p_C\n",
    "print(\"VC_chr2_count_C\", VC_chr2_count_C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "740e3270",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2987030\n",
      "746757.5 1493515.0 2240272.5 2987030.0\n"
     ]
    }
   ],
   "source": [
    "print(len(VC_chr1_fasta))\n",
    "x = 2987030/4\n",
    "y = 2987030/2\n",
    "z = 746757.5*3\n",
    "r = 746757.5*4\n",
    "print(x, y, z, r)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "a63f3a1a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2987030\n",
      "746757\n",
      "746757\n",
      "746757\n",
      "746757\n",
      "ID: NZ_CP016324.1\n",
      "Name: NZ_CP016324.1\n",
      "Description: NZ_CP016324.1 Vibrio cholerae 2740-80 chromosome 1, complete sequence\n",
      "Number of features: 0\n",
      "Seq('GGCAAAATGTGTGAGGATCTTAGTTATCGGTCGAAAAATAATGTGAATAACTTA...TAA')\n",
      "ID: NZ_CP016324.1\n",
      "Name: NZ_CP016324.1\n",
      "Description: NZ_CP016324.1 Vibrio cholerae 2740-80 chromosome 1, complete sequence\n",
      "Number of features: 0\n",
      "Seq('GTCAGCGTCCCCGTTTTATCAAACACCACGGTATCTACCTGACTTGCGGTCTGC...CGC')\n",
      "ID: NZ_CP016324.1\n",
      "Name: NZ_CP016324.1\n",
      "Description: NZ_CP016324.1 Vibrio cholerae 2740-80 chromosome 1, complete sequence\n",
      "Number of features: 0\n",
      "Seq('TGATTACCCGACGTGGGACGATCAAGCGTTTATTTTCCGTGTGCCAACCCTGTA...GTA')\n",
      "ID: NZ_CP016324.1\n",
      "Name: NZ_CP016324.1\n",
      "Description: NZ_CP016324.1 Vibrio cholerae 2740-80 chromosome 1, complete sequence\n",
      "Number of features: 0\n",
      "Seq('CTCATGTTATTTTATGTTTTTTGAGATAAGATAACAGCCACGAATGTGGCTGTT...ACG')\n"
     ]
    }
   ],
   "source": [
    "print(len(VC_chr1_fasta))\n",
    "VC_chr1_1_4 = VC_chr1_fasta[:746757]\n",
    "VC_chr1_2_4 = VC_chr1_fasta[746758:1493515]\n",
    "VC_chr1_3_4 = VC_chr1_fasta[1493515:2240272]\n",
    "VC_chr1_4_4 = VC_chr1_fasta[2240273:]\n",
    "print(len(VC_chr1_1_4))\n",
    "print(len(VC_chr1_2_4))\n",
    "print(len(VC_chr1_3_4))\n",
    "print(len(VC_chr1_4_4))\n",
    "print(VC_chr1_1_4)\n",
    "print(VC_chr1_2_4)\n",
    "print(VC_chr1_3_4)\n",
    "print(VC_chr1_4_4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "19a0a937",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VC_chr1_n_4_A\n",
      "186598\n",
      "197027\n",
      "200824\n",
      "197313\n",
      "VC_chr1_n_4_G\n",
      "192625\n",
      "186337\n",
      "167890\n",
      "167601\n",
      "VC_chr1_n_4_T\n",
      "193615\n",
      "196086\n",
      "194207\n",
      "192361\n",
      "VC_chr1_n_4_C\n",
      "173919\n",
      "167307\n",
      "183836\n",
      "189482\n"
     ]
    }
   ],
   "source": [
    "VC_chr1_1_4_A = VC_chr1_1_4.count(\"A\")\n",
    "VC_chr1_2_4_A = VC_chr1_2_4.count(\"A\")\n",
    "VC_chr1_3_4_A = VC_chr1_3_4.count(\"A\")\n",
    "VC_chr1_4_4_A = VC_chr1_4_4.count(\"A\")\n",
    "print('VC_chr1_n_4_A')\n",
    "print(VC_chr1_1_4_A) \n",
    "print(VC_chr1_2_4_A)\n",
    "print(VC_chr1_3_4_A) \n",
    "print(VC_chr1_4_4_A) \n",
    "VC_chr1_1_4_G = VC_chr1_1_4.count(\"G\")\n",
    "VC_chr1_2_4_G = VC_chr1_2_4.count(\"G\")\n",
    "VC_chr1_3_4_G = VC_chr1_3_4.count(\"G\")\n",
    "VC_chr1_4_4_G = VC_chr1_4_4.count(\"G\")\n",
    "print('VC_chr1_n_4_G')\n",
    "print(VC_chr1_1_4_G) \n",
    "print(VC_chr1_2_4_G)\n",
    "print(VC_chr1_3_4_G) \n",
    "print(VC_chr1_4_4_G) \n",
    "VC_chr1_1_4_T = VC_chr1_1_4.count('T')\n",
    "VC_chr1_2_4_T = VC_chr1_2_4.count(\"T\")\n",
    "VC_chr1_3_4_T = VC_chr1_3_4.count(\"T\")\n",
    "VC_chr1_4_4_T = VC_chr1_4_4.count(\"T\")\n",
    "print('VC_chr1_n_4_T')\n",
    "print(VC_chr1_1_4_T) \n",
    "print(VC_chr1_2_4_T)\n",
    "print(VC_chr1_3_4_T) \n",
    "print(VC_chr1_4_4_T)\n",
    "VC_chr1_1_4_C = VC_chr1_1_4.count('C')\n",
    "VC_chr1_2_4_C = VC_chr1_2_4.count(\"C\")\n",
    "VC_chr1_3_4_C = VC_chr1_3_4.count(\"C\")\n",
    "VC_chr1_4_4_C = VC_chr1_4_4.count(\"C\")\n",
    "print('VC_chr1_n_4_C')\n",
    "print(VC_chr1_1_4_C) \n",
    "print(VC_chr1_2_4_C)\n",
    "print(VC_chr1_3_4_C) \n",
    "print(VC_chr1_4_4_C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "ca9cac12",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2897536\n",
      "724384.0 1448768.0 2173152 2987030.0\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "2897536"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(VF_chr1_fasta))\n",
    "x = 2897536/4\n",
    "y = 2897536/2\n",
    "z = 724384*3\n",
    "r = 746757.5*4\n",
    "print(x, y, z, r)\n",
    "724384*4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "6bd8b09c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2897536\n",
      "724384\n",
      "724384\n",
      "724384\n",
      "724384\n",
      "ID: NC_006840.2\n",
      "Name: NC_006840.2\n",
      "Description: NC_006840.2 Aliivibrio fischeri ES114 chromosome I, complete sequence\n",
      "Number of features: 0\n",
      "Seq('AAGATCACTTAATATATATAAGATCTTTTAAAGAGATCTTTTATTAGATCTATT...GAA')\n",
      "ID: NC_006840.2\n",
      "Name: NC_006840.2\n",
      "Description: NC_006840.2 Aliivibrio fischeri ES114 chromosome I, complete sequence\n",
      "Number of features: 0\n",
      "Seq('TAGAATAAGTCGTTCTATTTGTGCTGATTAAAAAAAAAGCAAAACAGTAAGGTT...TTT')\n",
      "ID: NC_006840.2\n",
      "Name: NC_006840.2\n",
      "Description: NC_006840.2 Aliivibrio fischeri ES114 chromosome I, complete sequence\n",
      "Number of features: 0\n",
      "Seq('TCAGAATGATTTTTCACAGACAATACCTAATTATGAAATAAAACGGTTGATACC...CGC')\n",
      "ID: NC_006840.2\n",
      "Name: NC_006840.2\n",
      "Description: NC_006840.2 Aliivibrio fischeri ES114 chromosome I, complete sequence\n",
      "Number of features: 0\n",
      "Seq('CACCACTTAATGCGACAATAAGATGTGGTTTTTTAACGGAATAAGATGTTATTT...AGA')\n"
     ]
    }
   ],
   "source": [
    "print(len(VF_chr1_fasta))\n",
    "VF_chr1_1_4 = VF_chr1_fasta[:724384]\n",
    "VF_chr1_2_4 = VF_chr1_fasta[724384:1448768]\n",
    "VF_chr1_3_4 = VF_chr1_fasta[1448768:2173152]\n",
    "VF_chr1_4_4 = VF_chr1_fasta[2173152:]\n",
    "print(len(VF_chr1_1_4))\n",
    "print(len(VF_chr1_2_4))\n",
    "print(len(VF_chr1_3_4))\n",
    "print(len(VF_chr1_4_4))\n",
    "print(VF_chr1_1_4)\n",
    "print(VF_chr1_2_4)\n",
    "print(VF_chr1_3_4)\n",
    "print(VF_chr1_4_4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "693379ae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VF_chr1_n_4_A\n",
      "216887\n",
      "221758\n",
      "224508\n",
      "219659\n",
      "VF_chr1_n_4_G\n",
      "153940\n",
      "148274\n",
      "127419\n",
      "133850\n",
      "VF_chr1_n_4_T\n",
      "222502\n",
      "226081\n",
      "222833\n",
      "214632\n",
      "VF_chr1_n_4_C\n",
      "131055\n",
      "128271\n",
      "149624\n",
      "156243\n"
     ]
    }
   ],
   "source": [
    "VF_chr1_1_4_A = VF_chr1_1_4.count(\"A\")\n",
    "VF_chr1_2_4_A = VF_chr1_2_4.count(\"A\")\n",
    "VF_chr1_3_4_A = VF_chr1_3_4.count(\"A\")\n",
    "VF_chr1_4_4_A = VF_chr1_4_4.count(\"A\")\n",
    "print('VF_chr1_n_4_A')\n",
    "print(VF_chr1_1_4_A) \n",
    "print(VF_chr1_2_4_A)\n",
    "print(VF_chr1_3_4_A) \n",
    "print(VF_chr1_4_4_A) \n",
    "VF_chr1_1_4_G = VF_chr1_1_4.count(\"G\")\n",
    "VF_chr1_2_4_G = VF_chr1_2_4.count(\"G\")\n",
    "VF_chr1_3_4_G = VF_chr1_3_4.count(\"G\")\n",
    "VF_chr1_4_4_G = VF_chr1_4_4.count(\"G\")\n",
    "print('VF_chr1_n_4_G')\n",
    "print(VF_chr1_1_4_G) \n",
    "print(VF_chr1_2_4_G)\n",
    "print(VF_chr1_3_4_G) \n",
    "print(VF_chr1_4_4_G) \n",
    "VF_chr1_1_4_T = VF_chr1_1_4.count('T')\n",
    "VF_chr1_2_4_T = VF_chr1_2_4.count(\"T\")\n",
    "VF_chr1_3_4_T = VF_chr1_3_4.count(\"T\")\n",
    "VF_chr1_4_4_T = VF_chr1_4_4.count(\"T\")\n",
    "print('VF_chr1_n_4_T')\n",
    "print(VF_chr1_1_4_T) \n",
    "print(VF_chr1_2_4_T)\n",
    "print(VF_chr1_3_4_T) \n",
    "print(VF_chr1_4_4_T)\n",
    "VF_chr1_1_4_C = VF_chr1_1_4.count('C')\n",
    "VF_chr1_2_4_C = VF_chr1_2_4.count(\"C\")\n",
    "VF_chr1_3_4_C = VF_chr1_3_4.count(\"C\")\n",
    "VF_chr1_4_4_C = VF_chr1_4_4.count(\"C\")\n",
    "print('VF_chr1_n_4_C')\n",
    "print(VF_chr1_1_4_C) \n",
    "print(VF_chr1_2_4_C)\n",
    "print(VF_chr1_3_4_C) \n",
    "print(VF_chr1_4_4_C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c513159",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64032d6d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f882ed1b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7899c966",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b058a74",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "d5346662",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr = pd.read_csv('Chr.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "c8881f7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Lineage</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Site</th>\n",
       "      <th>AncestralBase</th>\n",
       "      <th>MutatedBase</th>\n",
       "      <th>CODorIG</th>\n",
       "      <th>SYNorNONSYN</th>\n",
       "      <th>AACode</th>\n",
       "      <th>AAChange</th>\n",
       "      <th>Location</th>\n",
       "      <th>...</th>\n",
       "      <th>Consensus</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>AncestralBase_2</th>\n",
       "      <th>MutatedBase_2</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutation2</th>\n",
       "      <th>A_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>C_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>415883</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>CCA&gt;TCA</td>\n",
       "      <td>P&gt;S</td>\n",
       "      <td>415649:416117</td>\n",
       "      <td>...</td>\n",
       "      <td>98.85</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>1271811</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>IG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "      <td>891597</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>GTG&gt;GGG</td>\n",
       "      <td>V&gt;G</td>\n",
       "      <td>890434:892516</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>254586</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>CGC&gt;TGC</td>\n",
       "      <td>R&gt;C</td>\n",
       "      <td>254016:255825</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>49</td>\n",
       "      <td>1</td>\n",
       "      <td>540987</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>COD</td>\n",
       "      <td>syn</td>\n",
       "      <td>AGT&gt;AGC</td>\n",
       "      <td>S&gt;S</td>\n",
       "      <td>540663:542025</td>\n",
       "      <td>...</td>\n",
       "      <td>98.67</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>491568</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>ACA&gt;GCA</td>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>491432:491978</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>30</td>\n",
       "      <td>2</td>\n",
       "      <td>77590</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>GCA&gt;ACA</td>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>76939:78583</td>\n",
       "      <td>...</td>\n",
       "      <td>98.79</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>232326</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>COD</td>\n",
       "      <td>syn</td>\n",
       "      <td>GCT&gt;GCC</td>\n",
       "      <td>A&gt;A</td>\n",
       "      <td>231905:233111</td>\n",
       "      <td>...</td>\n",
       "      <td>97.98</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>47</td>\n",
       "      <td>2</td>\n",
       "      <td>479984</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>COD</td>\n",
       "      <td>syn</td>\n",
       "      <td>CTG&gt;TTG</td>\n",
       "      <td>L&gt;L</td>\n",
       "      <td>479737:481600</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>22</td>\n",
       "      <td>2</td>\n",
       "      <td>187312</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>COD</td>\n",
       "      <td>syn</td>\n",
       "      <td>CAA&gt;CAG</td>\n",
       "      <td>Q&gt;Q</td>\n",
       "      <td>186771:188700</td>\n",
       "      <td>...</td>\n",
       "      <td>94.87</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Lineage  Chromosome     Site AncestralBase MutatedBase CODorIG  \\\n",
       "0          4           1   415883             G           A     COD   \n",
       "1         29           1  1271811             T           C      IG   \n",
       "2         53           1   891597             T           G     COD   \n",
       "3         39           1   254586             G           A     COD   \n",
       "4         49           1   540987             T           C     COD   \n",
       "..       ...         ...      ...           ...         ...     ...   \n",
       "995       24           2   491568             A           G     COD   \n",
       "996       30           2    77590             C           T     COD   \n",
       "997       33           2   232326             A           G     COD   \n",
       "998       47           2   479984             C           T     COD   \n",
       "999       22           2   187312             T           C     COD   \n",
       "\n",
       "    SYNorNONSYN   AACode AAChange       Location  ... Consensus  \\\n",
       "0        nonsyn  CCA>TCA      P>S  415649:416117  ...     98.85   \n",
       "1             -        -        -              -  ...    100.00   \n",
       "2        nonsyn  GTG>GGG      V>G  890434:892516  ...    100.00   \n",
       "3        nonsyn  CGC>TGC      R>C  254016:255825  ...    100.00   \n",
       "4           syn  AGT>AGC      S>S  540663:542025  ...     98.67   \n",
       "..          ...      ...      ...            ...  ...       ...   \n",
       "995      nonsyn  ACA>GCA      T>A  491432:491978  ...    100.00   \n",
       "996      nonsyn  GCA>ACA      A>T    76939:78583  ...     98.79   \n",
       "997         syn  GCT>GCC      A>A  231905:233111  ...     97.98   \n",
       "998         syn  CTG>TTG      L>L  479737:481600  ...    100.00   \n",
       "999         syn  CAA>CAG      Q>Q  186771:188700  ...     94.87   \n",
       "\n",
       "           Cell_line AncestralBase_2  MutatedBase_2  Mutation Mutation2  \\\n",
       "0     V. cholerae WT               G              A       G>A       G>A   \n",
       "1     V. cholerae WT               T              C       T>C       T>C   \n",
       "2     V. cholerae WT               T              G       T>G       T>G   \n",
       "3     V. cholerae WT               G              A       G>A       G>A   \n",
       "4     V. cholerae WT               T              C       T>C       T>C   \n",
       "..               ...             ...            ...       ...       ...   \n",
       "995  V. cholerae mut               A              G       A>G       A>G   \n",
       "996  V. cholerae mut               C              T       C>T       C>T   \n",
       "997  V. cholerae mut               A              G       A>G       A>G   \n",
       "998  V. cholerae mut               C              T       C>T       C>T   \n",
       "999  V. cholerae mut               T              C       T>C       T>C   \n",
       "\n",
       "    A_count T_count G_count C_count  \n",
       "0    769143  788889  751561  677437  \n",
       "1    769143  788889  751561  677437  \n",
       "2    769143  788889  751561  677437  \n",
       "3    769143  788889  751561  677437  \n",
       "4    769143  788889  751561  677437  \n",
       "..      ...     ...     ...     ...  \n",
       "995  288245  297682  272044  243960  \n",
       "996  288245  297682  272044  243960  \n",
       "997  288245  297682  272044  243960  \n",
       "998  288245  297682  272044  243960  \n",
       "999  288245  297682  272044  243960  \n",
       "\n",
       "[1000 rows x 24 columns]"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr.head(1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "21e0d159",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_mut = Chr[(Chr[\"Chromosome\"] == 1)  & (Chr[\"Cell_line\"] == 'V. cholerae mut')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "70d58fb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_1_4_mut = Chr[(Chr[\"Chromosome\"] == 1)  & (Chr[\"Cell_line\"] == 'V. cholerae mut') & (Chr['Site'] < 7467578) ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60a7d3ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_2_4_mut = Chr[(Chr[\"Chromosome\"] == 1)  & (Chr[\"Cell_line\"] == 'V. cholerae mut') & (Chr['Site']>=7467578])& (Chr['Site'] < 1493516)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b84b3222",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_3_4_mut = Chr[(Chr[\"Chromosome\"] == 1)  & (Chr[\"Cell_line\"] == 'V. cholerae mut') & (Chr('Site')>=7467578:])& (Chr('Site') < 1493516)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8c0f10d",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_4_4_mut = Chr[(Chr[\"Chromosome\"] == 1)  & (Chr[\"Cell_line\"] == 'V. cholerae mut') & (Chr('Site')==[1493516:])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b1626e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "746757.5 1493515.0 2240272.5 2987030.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "403ab2f7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "8552a51b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "V. cholerae mut    777\n",
       "Name: Cell_line, dtype: int64"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut[\"Chromosome\"].value_counts()\n",
    "Chr_1_VC_mut[\"Cell_line\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "c9b8d7fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_mut_1_4 = Chr_1_VC_mut[Chr_1_VC_mut['Site'] < 746757.5]       \n",
    "Chr_1_VC_mut_2_4 = Chr_1_VC_mut[(Chr_1_VC_mut['Site'] > 746757.5) & (Chr_1_VC_mut['Site'] < 1493515)]\n",
    "Chr_1_VC_mut_3_4 = Chr_1_VC_mut[(Chr_1_VC_mut['Site'] > 1493515) & (Chr_1_VC_mut['Site'] < 2240272.5)]\n",
    "Chr_1_VC_mut_4_4 = Chr_1_VC_mut[Chr_1_VC_mut['Site'] > 2240272.5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "eaa7174d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "C>T    90\n",
       "A>G    73\n",
       "G>A    29\n",
       "T>C    26\n",
       "Name: Mutation2, dtype: int64"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_1_4['Mutation2'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "b024fce1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "A>G    70\n",
       "C>T    51\n",
       "T>C    31\n",
       "G>A    22\n",
       "G>C     2\n",
       "C>A     1\n",
       "A>T     1\n",
       "Name: Mutation2, dtype: int64"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_2_4['Mutation2'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "18e1d2e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "A>G    70\n",
       "C>T    51\n",
       "T>C    31\n",
       "G>A    22\n",
       "G>C     2\n",
       "C>A     1\n",
       "A>T     1\n",
       "Name: Mutation2, dtype: int64"
      ]
     },
     "execution_count": 242,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_2_4['Mutation2'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "id": "ec845234",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "C>T    93\n",
       "A>G    60\n",
       "T>C    27\n",
       "G>A    23\n",
       "G>C     2\n",
       "C>G     1\n",
       "A>C     1\n",
       "C>A     1\n",
       "Name: Mutation2, dtype: int64"
      ]
     },
     "execution_count": 243,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_4_4['Mutation2'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "f9d65f11",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_WT = Chr[(Chr[\"Chromosome\"] == 1)  & (Chr[\"Cell_line\"] == 'V. cholerae WT')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "11ffd769",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_WT_1_4 = Chr_1_VC_WT[Chr_1_VC_WT['Site'] < 746757.5]       \n",
    "Chr_1_VC_WT_2_4 = Chr_1_VC_WT[(Chr_1_VC_WT['Site'] > 746757.5) & (Chr_1_VC_WT['Site'] < 1493515)]\n",
    "Chr_1_VC_WT_3_4 = Chr_1_VC_WT[(Chr_1_VC_WT['Site'] > 1493515) & (Chr_1_VC_WT['Site'] < 2240272.5)]\n",
    "Chr_1_VC_WT_4_4 = Chr_1_VC_WT[Chr_1_VC_WT['Site'] > 2240272.5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "171ab736",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base        Cell_line  A_count  G_count  T_count  C_count  \\\n",
       "0       A>C            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "1       A>G            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "2       A>T            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "3       C>A            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "4       C>G            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "5       C>T            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "6       G>A            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "7       G>C            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "8       G>T            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "9       T>A            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "10      T>C            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "11      T>G            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "\n",
       "    Nucleotide_count  Mut_count  Total generations  \n",
       "0                  0          0              27588  \n",
       "1                  0          0              27588  \n",
       "2                  0          0              27588  \n",
       "3                  0          0              27588  \n",
       "4                  0          0              27588  \n",
       "5                  0          0              27588  \n",
       "6                  0          0              27588  \n",
       "7                  0          0              27588  \n",
       "8                  0          0              27588  \n",
       "9                  0          0              27588  \n",
       "10                 0          0              27588  \n",
       "11                 0          0              27588  "
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_table = pd.DataFrame(\n",
    "    {\n",
    "        'Mutation': ['A>C', 'A>G', 'A>T', 'C>A', 'C>G', 'C>T', 'G>A', 'G>C', 'G>T', 'T>A', 'T>C', 'T>G'],\n",
    "        'Mutated_base': ['A', 'A', 'A', 'C', 'C', 'C', 'G', 'G', 'G', 'T', 'T' ,'T'],\n",
    "        'Cell_line': ['V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut','V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut'],\n",
    "        'A_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'G_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'T_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'C_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Nucleotide_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Mut_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Total generations' :[27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588]\n",
    "    }\n",
    ")\n",
    "Chr_1_VC_mut_table['A_count'] = VC_chr1_A_count\n",
    "Chr_1_VC_mut_table['G_count'] = VC_chr1_G_count\n",
    "Chr_1_VC_mut_table['T_count'] = VC_chr1_T_count\n",
    "Chr_1_VC_mut_table['C_count'] = VC_chr1_C_count\n",
    "Chr_1_VC_mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d04dcb8f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "9e4d0b77",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>781763</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>781763</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>781763</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714544</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714454</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714454</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714454</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>776269</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>776269</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>776269</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base        Cell_line  A_count  G_count  T_count  C_count  \\\n",
       "0       A>C            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "1       A>G            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "2       A>T            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "3       C>A            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "4       C>G            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "5       C>T            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "6       G>A            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "7       G>C            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "8       G>T            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "9       T>A            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "10      T>C            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "11      T>G            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "\n",
       "    Nucleotide_count  Mut_count  Total generations  \n",
       "0             781763          0              27588  \n",
       "1             781763          0              27588  \n",
       "2             781763          0              27588  \n",
       "3             714544          0              27588  \n",
       "4             714544          0              27588  \n",
       "5             714544          0              27588  \n",
       "6             714454          0              27588  \n",
       "7             714454          0              27588  \n",
       "8             714454          0              27588  \n",
       "9             776269          0              27588  \n",
       "10            776269          0              27588  \n",
       "11            776269          0              27588  "
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def Nuc_count(n):\n",
    "    if n == 'A':\n",
    "        return VC_chr1_A_count\n",
    "    elif n == 'C':\n",
    "        return VC_chr1_C_count\n",
    "    elif n == 'G':\n",
    "        return VC_chr1_G_count\n",
    "    else:\n",
    "        return VC_chr1_T_count\n",
    "\n",
    "Chr_1_VC_mut_table['Nucleotide_count'] = Chr_1_VC_mut_table['Mutated_base'].map(Nuc_count)\n",
    "Chr_1_VC_mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "7f48e53f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Mut_count(n):\n",
    "    if n == 'A>C':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['A>C']\n",
    "    elif n == 'A>G':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['A>G']\n",
    "    elif n == 'A>T':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['A>T']\n",
    "    elif n == 'C>A':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['C>A']\n",
    "    elif n == 'C>G':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['C>G']\n",
    "    elif n == 'C>T':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['C>T']\n",
    "    elif n == 'G>A':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['G>A']\n",
    "    elif n == 'G>C':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['G>C']\n",
    "    elif n == 'G>T':    \n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['G>T']\n",
    "    elif n == 'T>A':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['T>A']\n",
    "    elif n == 'T>C':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['T>C']\n",
    "    elif:\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['T>G']\n",
    "    else:\n",
    "        return 0\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "id": "b1d75629",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>781763</td>\n",
       "      <td>2</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>781763</td>\n",
       "      <td>256</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>781763</td>\n",
       "      <td>2</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714544</td>\n",
       "      <td>2</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714544</td>\n",
       "      <td>1</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714544</td>\n",
       "      <td>299</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714454</td>\n",
       "      <td>94</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714454</td>\n",
       "      <td>4</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>714454</td>\n",
       "      <td>2</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>776269</td>\n",
       "      <td>1</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>776269</td>\n",
       "      <td>113</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>781763</td>\n",
       "      <td>714454</td>\n",
       "      <td>776269</td>\n",
       "      <td>714544</td>\n",
       "      <td>776269</td>\n",
       "      <td>1</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base        Cell_line  A_count  G_count  T_count  C_count  \\\n",
       "0       A>C            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "1       A>G            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "2       A>T            A  V. cholerae mut   781763   714454   776269   714544   \n",
       "3       C>A            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "4       C>G            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "5       C>T            C  V. cholerae mut   781763   714454   776269   714544   \n",
       "6       G>A            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "7       G>C            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "8       G>T            G  V. cholerae mut   781763   714454   776269   714544   \n",
       "9       T>A            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "10      T>C            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "11      T>G            T  V. cholerae mut   781763   714454   776269   714544   \n",
       "\n",
       "    Nucleotide_count  Mut_count  Total generations  \n",
       "0             781763          2              27588  \n",
       "1             781763        256              27588  \n",
       "2             781763          2              27588  \n",
       "3             714544          2              27588  \n",
       "4             714544          1              27588  \n",
       "5             714544        299              27588  \n",
       "6             714454         94              27588  \n",
       "7             714454          4              27588  \n",
       "8             714454          2              27588  \n",
       "9             776269          1              27588  \n",
       "10            776269        113              27588  \n",
       "11            776269          1              27588  "
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_table['Mut_count'] = Chr_1_VC_mut_table['Mutation'].map(Mut_count)\n",
    "Chr_1_VC_mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "id": "80a5f9ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Axes: xlabel='Mutation'>"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAiMAAAHSCAYAAAA+DMuQAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAA5nUlEQVR4nO3de1iUdeL//9eAggcENOVkGJ7Pqdlq1KfWSkMzNrPNYymYbppsB8pNykPaJtmqS58yzVKwNkvXNbLso1+XItPcXE+Vm+dDuq2AZoqiocL794c/p50AcxR4z43Px3XNdXmfX8M9cL+873tmXMYYIwAAAEv8bAcAAABXNsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsMpRZWTVqlWKj49XVFSUXC6XMjMzK3R7RUVFGj9+vBo3bqyaNWuqadOmeu6558Qn6AMAUH6q2Q7gjYKCAnXo0EHDhg1T3759K3x7U6dO1axZszR//ny1bdtW69evV2JiokJCQvTII49U+PYBALgSOKqM9OrVS7169SpzemFhoZ555hm98847Onr0qNq1a6epU6eqW7dul7S9zz//XHfffbd69+4tSYqJidE777yjdevWXdL6AABASY66TPNLkpKStHbtWr377rv66quvdN9996lnz57auXPnJa3vxhtvVFZWlnbs2CFJ+vLLL7V69eoLFiIAAOAdR50ZuZD9+/crPT1d+/fvV1RUlCTpySef1PLly5Wenq4pU6Z4vc6xY8cqPz9frVq1kr+/v4qKivT8889r8ODB5R0fAIArVpU5M/L111+rqKhILVq0UFBQkPvx6aefavfu3ZKkbdu2yeVyXfAxduxY9zoXLVqkt99+WwsWLNDGjRs1f/58TZs2TfPnz7f1NAEAqHKqzJmREydOyN/fXxs2bJC/v7/HtKCgIElSkyZNtHXr1guu56qrrnL/e8yYMRo7dqwGDBggSWrfvr2+/fZbpaamaujQoeX8DAAAuDJVmTLSqVMnFRUVKS8vTzfffHOp8wQEBKhVq1YXvc6TJ0/Kz8/z5JG/v7+Ki4svKysAAPiJo8rIiRMntGvXLvfw3r17tXnzZtWrV08tWrTQ4MGDNWTIEE2fPl2dOnXSoUOHlJWVpWuvvdb9jhhvxMfH6/nnn1ejRo3Utm1bbdq0STNmzNCwYcPK82kBAHBFcxkHfYJXdna2br311hLjhw4dqoyMDJ05c0Z//OMf9eabb+q7775T/fr1dcMNN2jSpElq376919s7fvy4xo8fr/fee095eXmKiorSwIEDNWHCBAUEBJTHUwIA4IrnqDICAACqnirzbhoAAOBMlBEAAGCVI25gLS4u1n/+8x/VqVNHLpfLdhwAAHARjDE6fvy4oqKiSrw79b85ooz85z//UXR0tO0YAADgEhw4cEBXX311mdMdUUbq1Kkj6dyTCQ4OtpwGAABcjPz8fEVHR7uP42VxRBk5f2kmODiYMgIAgMP80i0W3MAKAACsoowAAACrKCMAAMAqR9wzAgBwtqKiIp05c8Z2DJSz6tWry9/f/7LXQxkBAFQYY4xycnJ09OhR21FQQUJDQxUREXFZnwNGGQEAVJjzRSQsLEy1atXigyurEGOMTp48qby8PElSZGTkJa+LMgIAqBBFRUXuInLVVVfZjoMKULNmTUlSXl6ewsLCLvmSDTewAgAqxPl7RGrVqmU5CSrS+f17OfcEUUYAABWKSzNVW3nsX8oIAACwijICAACs4gZWAEClixm7rNK2te+F3l4vk5CQoPnz55cYv3PnTjVr1qw8YuG/UEYAAChFz549lZ6e7jGuQYMGHsOnT59WQEBAZcaqkrhMAwBAKQIDAxUREeHxuP3225WUlKTHHntM9evXV1xcnCRpy5Yt6tWrl4KCghQeHq4HHnhAhw8fdq+roKBAQ4YMUVBQkCIjIzV9+nR169ZNjz32mHsel8ulzMxMjwyhoaHKyMhwDx84cED9+vVTaGio6tWrp7vvvlv79u1zT09ISFCfPn00bdo0RUZG6qqrrtLo0aM93ulSWFiop556StHR0QoMDFSzZs00d+5cGWPUrFkzTZs2zSPD5s2b5XK5tGvXrsv/oZaBMgIAgBfmz5+vgIAArVmzRrNnz9bRo0d12223qVOnTlq/fr2WL1+u3Nxc9evXz73MmDFj9Omnn+r999/X//t//0/Z2dnauHGjV9s9c+aM4uLiVKdOHX322Wdas2aNgoKC1LNnT50+fdo93yeffKLdu3frk08+0fz585WRkeFRaIYMGaJ33nlH//u//6utW7fqtddeU1BQkFwul4YNG1bibFB6erpuueWWCr08xWUaAJLK/xr+pVynB3zJhx9+qKCgIPdwr169JEnNmzfXiy++6B7/xz/+UZ06ddKUKVPc4+bNm6fo6Gjt2LFDUVFRmjt3rv7yl7/o9ttvl3Su0Fx99dVe5Vm4cKGKi4v1xhtvuN9Om56ertDQUGVnZ+uOO+6QJNWtW1evvPKK/P391apVK/Xu3VtZWVkaMWKEduzYoUWLFmnlypXq3r27JKlJkybubSQkJGjChAlat26dunTpojNnzmjBggUlzpaUN8oIAACluPXWWzVr1iz3cO3atTVw4EB17tzZY74vv/xSn3zyiUdxOW/37t06deqUTp8+ra5du7rH16tXTy1btvQqz5dffqldu3apTp06HuN//PFH7d692z3ctm1bj09CjYyM1Ndffy3p3CUXf39//frXvy51G1FRUerdu7fmzZunLl266IMPPlBhYaHuu+8+r7J6izICAEApateuXeqlidq1a3sMnzhxQvHx8Zo6dWqJeSMjIy/6XguXyyVjjMe4/77X48SJE+rcubPefvvtEsv+94211atXL7He4uJiST99fPuFDB8+XA888ID+/Oc/Kz09Xf3796/wT9GljAAAcBmuu+46/e1vf1NMTIyqVSt5WG3atKmqV6+uL774Qo0aNZIk/fDDD9qxY4fHGYoGDRro4MGD7uGdO3fq5MmTHttZuHChwsLCFBwcfElZ27dvr+LiYn366afuyzQ/d+edd6p27dqaNWuWli9frlWrVl3StrzBDawAAFyG0aNH68iRIxo4cKD++c9/avfu3VqxYoUSExNVVFSkoKAgPfjggxozZow+/vhjbdmyRQkJCfLz8zwE33bbbXrllVe0adMmrV+/XiNHjvQ4yzF48GDVr19fd999tz777DPt3btX2dnZeuSRR/Tvf//7orLGxMRo6NChGjZsmDIzM93rWLRokXsef39/JSQkKCUlRc2bN1dsbGz5/KAugDMjAIBKV5VucI6KitKaNWv01FNP6Y477lBhYaGuueYa9ezZ0104/vSnP7kv59SpU0dPPPGEjh075rGe6dOnKzExUTfffLOioqL00ksvacOGDe7ptWrV0qpVq/TUU0+pb9++On78uBo2bKjbb7/dqzMls2bN0tNPP62HH35Y33//vRo1aqSnn37aY54HH3xQU6ZMUWJi4mX8ZC6ey/z8ApUPys/PV0hIiI4dO3bJp6YAXBjvpkF5+/HHH7V37141btxYNWrUsB3H53Tr1k0dO3ZUWlqa7SglfPbZZ7r99tt14MABhYeHX3DeC+3niz1+c2YEAABIOveBaIcOHdKzzz6r++677xeLSHnhnhEAACBJeuedd3TNNdfo6NGjHp+lUtE4MwIAgAXZ2dm2I5SQkJCghISESt8uZ0YAAIBVlBEAQIVywPskcBnKY/9SRgAAFeL8Z2T89wd3oeo5v39//smv3uCeEQBAhfD391doaKjy8vIknfucjPNf8AbnM8bo5MmTysvLU2hoqMf34XiLMgIAqDARERGS5C4kqHpCQ0Pd+/lSUUYAABXG5XIpMjJSYWFhHl/6hqqhevXql3VG5DzKCACgwvn7+5fLQQtVEzewAgAAqygjAADAKsoIAACwijICAACsoowAAACrvC4jq1atUnx8vKKiouRyuZSZmXnRy65Zs0bVqlVTx44dvd0sAACoorwuIwUFBerQoYNmzpzp1XJHjx7VkCFDdPvtt3u7SQAAUIV5/TkjvXr1Uq9evbze0MiRIzVo0CD5+/t7dTYFAABUbZVyz0h6err27NmjiRMnVsbmAACAg1T4J7Du3LlTY8eO1WeffaZq1S5uc4WFhSosLHQP5+fnV1Q8AABgWYWeGSkqKtKgQYM0adIktWjR4qKXS01NVUhIiPsRHR1dgSkBAIBNFVpGjh8/rvXr1yspKUnVqlVTtWrVNHnyZH355ZeqVq2aPv7441KXS0lJ0bFjx9yPAwcOVGRMAABgUYVepgkODtbXX3/tMe7VV1/Vxx9/rMWLF6tx48alLhcYGKjAwMCKjAYAAHyE12XkxIkT2rVrl3t479692rx5s+rVq6dGjRopJSVF3333nd588035+fmpXbt2HsuHhYWpRo0aJcYDAIArk9dlZP369br11lvdw8nJyZKkoUOHKiMjQwcPHtT+/fvLLyEAAKjSXMYYYzvEL8nPz1dISIiOHTum4OBg23GAKilm7LJyXd++F3qX6/oAOM/FHr/5bhoAAGAVZQQAAFhFGQEAAFZRRgAAgFWUEQAAYBVlBAAAWEUZAQAAVlFGAACAVZQRAABgFWUEAABYRRkBAABWUUYAAIBVlBEAAGAVZQQAAFhFGQEAAFZRRgAAgFWUEQAAYBVlBAAAWEUZAQAAVlFGAACAVZQRAABgFWUEAABYRRkBAABWUUYAAIBVlBEAAGAVZQQAAFhFGQEAAFZRRgAAgFWUEQAAYBVlBAAAWEUZAQAAVlFGAACAVZQRAABgFWUEAABYRRkBAABWUUYAAIBVXpeRVatWKT4+XlFRUXK5XMrMzLzg/EuWLFGPHj3UoEEDBQcHKzY2VitWrLjUvAAAoIrxuowUFBSoQ4cOmjlz5kXNv2rVKvXo0UMfffSRNmzYoFtvvVXx8fHatGmT12EBAEDVU83bBXr16qVevXpd9PxpaWkew1OmTNH777+vDz74QJ06dfJ28wAAoIrxuoxcruLiYh0/flz16tUrc57CwkIVFha6h/Pz8ysjGgAAsKDSb2CdNm2aTpw4oX79+pU5T2pqqkJCQtyP6OjoSkwIAAAqU6WWkQULFmjSpElatGiRwsLCypwvJSVFx44dcz8OHDhQiSkBAEBlqrTLNO+++66GDx+uv/71r+revfsF5w0MDFRgYGAlJQMAADZVShl55513NGzYML377rvq3bt3ZWzSp8WMXVbu69z3Aj9XAIAzeV1GTpw4oV27drmH9+7dq82bN6tevXpq1KiRUlJS9N133+nNN9+UdO7SzNChQ/XSSy+pa9euysnJkSTVrFlTISEh5fQ0AACAU3l9z8j69evVqVMn99tyk5OT1alTJ02YMEGSdPDgQe3fv989/5w5c3T27FmNHj1akZGR7sejjz5aTk8BAAA4mddnRrp16yZjTJnTMzIyPIazs7O93QQAALiC8N00AADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArPK6jKxatUrx8fGKioqSy+VSZmbmLy6TnZ2t6667ToGBgWrWrJkyMjIuISoAAKiKvC4jBQUF6tChg2bOnHlR8+/du1e9e/fWrbfeqs2bN+uxxx7T8OHDtWLFCq/DAgCAqqeatwv06tVLvXr1uuj5Z8+ercaNG2v69OmSpNatW2v16tX685//rLi4OG83DwAAqpgKv2dk7dq16t69u8e4uLg4rV27tqI3DQAAHMDrMyPeysnJUXh4uMe48PBw5efn69SpU6pZs2aJZQoLC1VYWOgezs/Pr+iYAADAEp98N01qaqpCQkLcj+joaNuRAABABanwMhIREaHc3FyPcbm5uQoODi71rIgkpaSk6NixY+7HgQMHKjomAACwpMIv08TGxuqjjz7yGLdy5UrFxsaWuUxgYKACAwMrOhoAAPABXp8ZOXHihDZv3qzNmzdLOvfW3c2bN2v//v2Szp3VGDJkiHv+kSNHas+ePfrDH/6gbdu26dVXX9WiRYv0+OOPl88zAAAAjuZ1GVm/fr06deqkTp06SZKSk5PVqVMnTZgwQZJ08OBBdzGRpMaNG2vZsmVauXKlOnTooOnTp+uNN97gbb0AAEDSJVym6datm4wxZU4v7dNVu3Xrpk2bNnm7KQAAcAXwyXfTAACAKwdlBAAAWEUZAQAAVlFGAACAVZQRAABgFWUEAABYRRkBAABWUUYAAIBVlBEAAGAVZQQAAFhFGQEAAFZRRgAAgFWUEQAAYBVlBAAAWEUZAQAAVlFGAACAVZQRAABgFWUEAABYRRkBAABWUUYAAIBVlBEAAGAVZQQAAFhFGQEAAFZRRgAAgFWUEQAAYBVlBAAAWEUZAQAAVlFGAACAVZQRAABgFWUEAABYRRkBAABWUUYAAIBVlBEAAGAVZQQAAFhFGQEAAFZRRgAAgFWXVEZmzpypmJgY1ahRQ127dtW6desuOH9aWppatmypmjVrKjo6Wo8//rh+/PHHSwoMAACqFq/LyMKFC5WcnKyJEydq48aN6tChg+Li4pSXl1fq/AsWLNDYsWM1ceJEbd26VXPnztXChQv19NNPX3Z4AADgfF6XkRkzZmjEiBFKTExUmzZtNHv2bNWqVUvz5s0rdf7PP/9cN910kwYNGqSYmBjdcccdGjhw4C+eTQEAAFcGr8rI6dOntWHDBnXv3v2nFfj5qXv37lq7dm2py9x4443asGGDu3zs2bNHH330ke68884yt1NYWKj8/HyPBwAAqJqqeTPz4cOHVVRUpPDwcI/x4eHh2rZtW6nLDBo0SIcPH9b//M//yBijs2fPauTIkRe8TJOamqpJkyZ5Ew0AADhUhb+bJjs7W1OmTNGrr76qjRs3asmSJVq2bJmee+65MpdJSUnRsWPH3I8DBw5UdEwAAGCJV2dG6tevL39/f+Xm5nqMz83NVURERKnLjB8/Xg888ICGDx8uSWrfvr0KCgr0u9/9Ts8884z8/Er2ocDAQAUGBnoTDQAAOJRXZ0YCAgLUuXNnZWVluccVFxcrKytLsbGxpS5z8uTJEoXD399fkmSM8TYvAACoYrw6MyJJycnJGjp0qK6//np16dJFaWlpKigoUGJioiRpyJAhatiwoVJTUyVJ8fHxmjFjhjp16qSuXbtq165dGj9+vOLj492lBAAAXLm8LiP9+/fXoUOHNGHCBOXk5Khjx45avny5+6bW/fv3e5wJGTdunFwul8aNG6fvvvtODRo0UHx8vJ5//vnyexYAAMCxXMYB10ry8/MVEhKiY8eOKTg42HacyxYzdlm5r3PfC73LfZ24spT365LXJICLPX7z3TQAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsuqQyMnPmTMXExKhGjRrq2rWr1q1bd8H5jx49qtGjRysyMlKBgYFq0aKFPvroo0sKDAAAqpZq3i6wcOFCJScna/bs2eratavS0tIUFxen7du3KywsrMT8p0+fVo8ePRQWFqbFixerYcOG+vbbbxUaGloe+QEAgMN5XUZmzJihESNGKDExUZI0e/ZsLVu2TPPmzdPYsWNLzD9v3jwdOXJEn3/+uapXry5JiomJubzUAACgyvDqMs3p06e1YcMGde/e/acV+Pmpe/fuWrt2banLLF26VLGxsRo9erTCw8PVrl07TZkyRUVFRWVup7CwUPn5+R4PAABQNXlVRg4fPqyioiKFh4d7jA8PD1dOTk6py+zZs0eLFy9WUVGRPvroI40fP17Tp0/XH//4xzK3k5qaqpCQEPcjOjram5gAAMBBKvzdNMXFxQoLC9OcOXPUuXNn9e/fX88884xmz55d5jIpKSk6duyY+3HgwIGKjgkAACzx6p6R+vXry9/fX7m5uR7jc3NzFRERUeoykZGRql69uvz9/d3jWrdurZycHJ0+fVoBAQEllgkMDFRgYKA30QAAgEN5dWYkICBAnTt3VlZWlntccXGxsrKyFBsbW+oyN910k3bt2qXi4mL3uB07digyMrLUIgIAAK4sXl+mSU5O1uuvv6758+dr69atGjVqlAoKCtzvrhkyZIhSUlLc848aNUpHjhzRo48+qh07dmjZsmWaMmWKRo8eXX7PAgAAOJbXb+3t37+/Dh06pAkTJignJ0cdO3bU8uXL3Te17t+/X35+P3Wc6OhorVixQo8//riuvfZaNWzYUI8++qieeuqp8nsWAADAsbwuI5KUlJSkpKSkUqdlZ2eXGBcbG6t//OMfl7IpAABQxfHdNAAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALDqkj4OHgAASDFjl5X7Ove90Lvc1+nrODMCAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKr6bBoBj8D0gQNXEmREAAGAVZQQAAFhFGQEAAFZRRgAAgFWUEQAAYBVlBAAAWEUZAQAAVlFGAACAVZQRAABgFWUEAABYRRkBAABWUUYAAIBVl1RGZs6cqZiYGNWoUUNdu3bVunXrLmq5d999Vy6XS3369LmUzQIAgCrI6zKycOFCJScna+LEidq4caM6dOiguLg45eXlXXC5ffv26cknn9TNN998yWEBAEDV43UZmTFjhkaMGKHExES1adNGs2fPVq1atTRv3rwylykqKtLgwYM1adIkNWnS5LICAwCAqsWrMnL69Glt2LBB3bt3/2kFfn7q3r271q5dW+ZykydPVlhYmB588MFLTwoAAKqkat7MfPjwYRUVFSk8PNxjfHh4uLZt21bqMqtXr9bcuXO1efPmi95OYWGhCgsL3cP5+fnexAQAAA5Soe+mOX78uB544AG9/vrrql+//kUvl5qaqpCQEPcjOjq6AlMCAACbvDozUr9+ffn7+ys3N9djfG5uriIiIkrMv3v3bu3bt0/x8fHuccXFxec2XK2atm/frqZNm5ZYLiUlRcnJye7h/Px8CgkAAFWUV2UkICBAnTt3VlZWlvvtucXFxcrKylJSUlKJ+Vu1aqWvv/7aY9y4ceN0/PhxvfTSS2UWjMDAQAUGBnoTDQAAOJRXZUSSkpOTNXToUF1//fXq0qWL0tLSVFBQoMTEREnSkCFD1LBhQ6WmpqpGjRpq166dx/KhoaGSVGI8AAC4MnldRvr3769Dhw5pwoQJysnJUceOHbV8+XL3Ta379++Xnx8f7AoAAC6O12VEkpKSkkq9LCNJ2dnZF1w2IyPjUjYJAACqKE5hAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArLqkMjJz5kzFxMSoRo0a6tq1q9atW1fmvK+//rpuvvlm1a1bV3Xr1lX37t0vOD8AALiyeF1GFi5cqOTkZE2cOFEbN25Uhw4dFBcXp7y8vFLnz87O1sCBA/XJJ59o7dq1io6O1h133KHvvvvussMDAADn87qMzJgxQyNGjFBiYqLatGmj2bNnq1atWpo3b16p87/99tt6+OGH1bFjR7Vq1UpvvPGGiouLlZWVddnhAQCA83lVRk6fPq0NGzaoe/fuP63Az0/du3fX2rVrL2odJ0+e1JkzZ1SvXr0y5yksLFR+fr7HAwAAVE1elZHDhw+rqKhI4eHhHuPDw8OVk5NzUet46qmnFBUV5VFofi41NVUhISHuR3R0tDcxAQCAg1Tqu2leeOEFvfvuu3rvvfdUo0aNMudLSUnRsWPH3I8DBw5UYkoAAFCZqnkzc/369eXv76/c3FyP8bm5uYqIiLjgstOmTdMLL7ygv//977r22msvOG9gYKACAwO9iQYAABzKqzMjAQEB6ty5s8fNp+dvRo2NjS1zuRdffFHPPfecli9fruuvv/7S0wIAgCrHqzMjkpScnKyhQ4fq+uuvV5cuXZSWlqaCggIlJiZKkoYMGaKGDRsqNTVVkjR16lRNmDBBCxYsUExMjPvekqCgIAUFBZXjUwEAAE7kdRnp37+/Dh06pAkTJignJ0cdO3bU8uXL3Te17t+/X35+P51wmTVrlk6fPq3f/va3HuuZOHGinn322ctLDwAAHM/rMiJJSUlJSkpKKnVadna2x/C+ffsuZRMAAOAKwXfTAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsIoyAgAArKKMAAAAqygjAADAKsoIAACwijICAACsoowAAACrKCMAAMAqyggAALCKMgIAAKyijAAAAKsoIwAAwCrKCAAAsKrapSw0c+ZM/elPf1JOTo46dOigl19+WV26dClz/r/+9a8aP3689u3bp+bNm2vq1Km68847Lzk0APiqmLHLyn2d+17oXe7rBHyJ12dGFi5cqOTkZE2cOFEbN25Uhw4dFBcXp7y8vFLn//zzzzVw4EA9+OCD2rRpk/r06aM+ffpoy5Ytlx0eAAA4n9dnRmbMmKERI0YoMTFRkjR79mwtW7ZM8+bN09ixY0vM/9JLL6lnz54aM2aMJOm5557TypUr9corr2j27NmXGb8k/ldSfsr7Z1kRP0f2N3BpnPD7jSuHV2Xk9OnT2rBhg1JSUtzj/Pz81L17d61du7bUZdauXavk5GSPcXFxccrMzPQ+LeBAFCYAuDCvysjhw4dVVFSk8PBwj/Hh4eHatm1bqcvk5OSUOn9OTk6Z2yksLFRhYaF7+NixY5Kk/Pz8X8xYXHjyF+fx1sVs1xtOyCiVf04nZJTY3+XFCRkl9nd5qYiM7SauKPd1bpkUV67rc8r+Lu+f5cX+HM8/F2PMBee7pBtYK1pqaqomTZpUYnx0dLSFNFJImpXNeoWM5ccJOclYfpyQk4zlxwk5q2LG48ePKyQkpMzpXpWR+vXry9/fX7m5uR7jc3NzFRERUeoyERERXs0vSSkpKR6XdoqLi3XkyBFdddVVcrlc3kQuVX5+vqKjo3XgwAEFBwdf9voqihNykrH8OCEnGcuPE3KSsfw4IWdFZDTG6Pjx44qKirrgfF6VkYCAAHXu3FlZWVnq06ePpHNFISsrS0lJSaUuExsbq6ysLD322GPucStXrlRsbGyZ2wkMDFRgYKDHuNDQUG+iXpTg4GCffVH8NyfkJGP5cUJOMpYfJ+QkY/lxQs7yznihMyLneX2ZJjk5WUOHDtX111+vLl26KC0tTQUFBe531wwZMkQNGzZUamqqJOnRRx/Vr3/9a02fPl29e/fWu+++q/Xr12vOnDnebhoAAFRBXpeR/v3769ChQ5owYYJycnLUsWNHLV++3H2T6v79++Xn99PHl9x4441asGCBxo0bp6efflrNmzdXZmam2rVrV37PAgAAONYl3cCalJRU5mWZ7OzsEuPuu+8+3XfffZeyqQoRGBioiRMnlrgU5GuckJOM5ccJOclYfpyQk4zlxwk5bWZ0mV96vw0AAEAF4ovyAACAVZQRAABgFWUEAABYRRkBADhecXGxPvzwQ9sxJJ378LDi4uIS44uKiirko96rgipfRnhRlI9hw4bp+PHjtmNctq1bt+rJJ5+0HcMRmjRpou+//952jMvC/r54Tt3fu3bt0tNPP62rr75a99xzj+04eu+993T99dfrxx9/LDHtxx9/1K9+9St98MEHFpL95OjRo5o1a5Z7ePDgwerbt6/7cd999+no0aOVmqlKlxEnvCgk33xh/Nz8+fN16tQpqxkuVUFBgebOnasbb7xRbdu21fLly21HcoR9+/apqKjIdgyv+dr+PnXqlJYuXVpqmc/Pz9fSpUs9vhjUFift71OnTunNN9/ULbfcopYtW+rzzz/XhAkT9O9//9t2NM2aNUt/+MMfVKtWrRLTateuraeeekqvvPKKhWQ/ef3117V69Wr38NKlS+Xn56eQkBCFhITo66+/VlpaWuWGMlVYjx49zOuvv17m9Llz55o77rijEhOV7sUXXzSDBg1yDwcFBZl7773XJCQkmISEBNOyZUszceJEewGNMS6Xy+Tm5lrN4K3Vq1ebxMREU7t2bePn52eeeOIJs3XrVtuxzK5du0xiYqJ7ODo62tStW9f9qF+/vtm2bZvFhOc4bZ/76v5OS0szt912W5nTb7/9dvPKK69UYqLSOWF/r1u3zvzud78zwcHBplOnTmbatGnG39/f/Otf/7IdzS0yMtLs3LmzzOk7d+40kZGRlZiopC5dupiVK1e6h4OCgszu3bvdw0uWLDEdO3as1Ew++a295WXLli169dVXy5x+yy23aNy4cZWYqHSLFy/W888/7zHuxRdfVJMmTSSdO8MzefJkPfvssxbS/eT48eOqUaPGBeex/Z0LeXl5ysjI0Lx583Ts2DENHDhQ2dnZio2N1bBhw9SqVSur+STp5Zdfdn9isST98MMPmjBhgsLCwiRJCxcu1J///GfNnj3bVkS3FStW/OL3SvzmN7+ppDQlOWF/v/322xo/fnyZ0x977DFNnjxZo0ePrsRUpfPl/X3ttdcqPz9fgwYN0ueff662bdtKksaOHWslT1l++OEHnT17tszpZ86c0Q8//FCJiUras2ePWrZs6R5u2bKlAgIC3MMdOnTQzp07KzVTlS4jTnhRSL75wihNixYtypxmjJHL5bJ+mveaa67Rb3/7W7300kvq0aOHx1cT+IqsrCzNnTvXY9y9997rLp8xMTEaPny4jWglDB069ILTbe9zJ+zvnTt3qkOHDmVOv/baa33i91vy7f29fft29e/fX7feeqvatGljJcPFiImJ0fr168sswuvXr9c111xTyak8FRQU6NixY4qOjnZn+vn00u61rEhVuow44UUh+eYLozSLFy9WvXr1bMe4oGuuuUarV69Wo0aNdM011/jE/4x/bt++fR5fpz18+HCP/43GxMT4xLVvScrJyXGfsfFFTtjfZ8+e1aFDh9SoUaNSpx86dOiC/2mqTL68v/fs2aOMjAyNGjVKp06d0sCBAzV48GC5XC7b0Tz07dtXzzzzjHr06OFxBlQ69/MdN26c7r//fkvpzmnSpIk2btxY5nfErV+/Xo0bN67cUJV6UaiSPf3006ZRo0YmJyenxLSDBw+aRo0amaefftpCMk9t27Y18+fPL3P6vHnzTJs2bSoxUUlOuJ583vl7B4KCgsx1111nZsyYYapVq2a++eYb29GMMcYEBwebL774oszpX3zxhalTp04lJiqdn5+fI/a5r+/vrl27mhdeeKHM6VOmTDFdu3atxESlc8r+NsaYrKwsM3jwYFOzZk3jcrnMmDFjzPbt223HMsYYk5+fb9q2bWvq1KljRo0aZdLS0kxaWpoZOXKkqVOnjmnTpo3Jz8+3mnHcuHEmOjq6zGNjdHS0eeaZZyo1U5UuI054URjjmy+Mn3NSGTnv+PHjZs6cOSY2Nta4XC7TrVs3M2fOHJOXl2c1V2xsrHn++efLnD558mQTGxtbiYlK57R97qv7+7XXXjO1a9c2H3zwQYlpS5cuNbVr1zavvfaahWSenLa/jTHm6NGjZubMmaZz587G5XKZ9u3b245kjDmXa9SoUaZevXrG5XIZl8tl6tata0aNGmWOHDliO57Jz883rVu3NnXq1DEPP/yw+9g4atQoU6dOHdOqVatKPzZW6TJijO+/KIzxzRfGz8XExJjDhw9bzXA5vvnmG/PEE0+YsLAwU61aNatZ5syZY2rVqmU+/PDDEtOWLl1qatWqZebMmWMhmaeEhATrr7tL5Uv72xhjBg8ebFwul2ndurXp06eP6dOnj2nVqpXx8/MzAwYMsB3PGOPs/W2MMZs2bTK///3vbcfwUFxcbPLy8kxubq4pLi62HcfDkSNHzEMPPWTq1q3rcWx86KGHzPfff1/pea6Yb+01xujw4cMyxqhBgwY+d53xhx9+UEpKihYtWuT+TJHQ0FD169dPU6ZM8fl7NZzi7NmzWrp0qfr27Ws1x8CBA7Vw4UK1atXKffPy9u3btX37dt17771atGiR1XxVha/sb0latGiRFixYoJ07d8oYoxYtWmjQoEHq16+f7Wi4ghljdOjQIUmye2ys9PqDCyouLja5ubk+2aT/2969e80999zjE5+H4VTvvPOOufvuu03r1q1N69atzW9+8xvzzjvv2I5Vpn/84x+mbt265uOPP7YdBZXA1/e3U/4GOSWnbb73PrgKtG/fPvXt21fbt2+3HaVMLpdLYWFhCgsL87mzN/9t/vz5yszMLPEWVV/zxRdfqF69evrkk09sRylhwIAByszM1DfffKNvvvlG77//vgYMGGA7VpkyMjJUUFCg9PR021HK5Mv722l8fX875W+QE3L6xLHRdhuqTM8++6z7zmtf5oQm3aRJEzN48GATGRlpioqKbMcp08iRI01AQIB54IEHbEdxtB9//NHUq1fPTJ061dSuXdscP37cdqRSOWF/+/oZB2Ocsb+d8jfICTl94dh4RZURJ7wojPGNF8aFfPrppyYkJMTk5+eb8PBw89FHH9mOVCon/EE1xhkHp4ULF5pGjRqZ4uJi07ZtW5Oenm47UglO2d9OKEy+vr+d8jfIKTl94dh4xVymWbVqlb7//nvNmjVLxcXFWrFihe1IZXrzzTc1aNAg/eUvf/GJDzv7ufnz5+vee+9VnTp1NGDAAGVkZNiOVKr3339fQUFBGjNmjGJiYrR48WLbkUrl66fDpXP7/PwHTN1///0+mdUJ+7uwsFCLFi3Sc889pyVLlujEiRO2I5XK1/e3U/4GOSGnzxwbrVQgC4YNG2aGDRtmjDHm0UcfNf369bOcqHS+3qRPnjxpgoODTXZ2tjHGmA0bNpiaNWuao0ePWk5W0p133mlSUlKMMcakpqaaW265xXKikpzwv/mDBw+a6tWruy8b7t+/3/j7+5s9e/ZYTubJCfvb1884GOP7+9spf4OcktNXjo1XRBlxyovCGN95YZTlrbfeMjExMR7j2rZta2bPnm0pUel8/Q/qeU44OP3pT38yv/rVrzzG3XbbbebZZ5+1lKgkp+xvJxQmX9/fTvkb5IScvnRsvCLKiBNeFMb41gujLD169CjxabCpqanmhhtusJSodL7+B/U8Jxyc2rVrZ15++WWPcenp6aZJkyaWEpXkhP3tlMLk6/vbKX+DnJDTl46NV0QZccKLwhjfemGU5vwfzx07dniMP3DggPH39/eZ74Ywxvf/oBrjjIPT/v37za233lri03ePHz9ubr/99hKvBVucsL+dUJh8fX875W+QU3L60rGxypcRp7wojPGtF8al+Prrr21HMMb4/h/U85xwcHICp+xvJxQmXDl87dhY5cvIL/GlA6gvvTAuVn5+vnnttddMly5djJ+fn+04juLrB6esrCzTunVrc+zYsRLTjh49atq0aWNWrVplIZnzOKEwnTx50uPL/MaOHWsef/xx92PMmDHm1KlT1vJNmjTJFBQUWNv+xXJKTl9zRZYRJx5AfaU0nffpp5+aIUOGmNq1a5vmzZubp556yqxbt852LMccQJ1wcIqPjzczZswoc/pLL71k+vTpU4mJSvL1A6iTzJo1y9x1113u4aCgINO1a1fTrVs3061bNxMREXHB10NF8/Pzc8S3Cjshpy8WpiuqjPjqAbQsvlaaDh48aFJTU02zZs1MWFiYSUpKMtWqVTP/+te/bEdzc8IB1CkaNWpkvvnmmzKnb9261URHR1diopJ8/QBqjHMK0//8z/+YpUuXuoeDgoLM7t273cNvvfWW1UvGLpfL5w/yxjgjpy8WpipfRpxwAP05XyxNd911lwkODjYDBw40H374oTl79qwxxvjcz9IJB1CnnL0JDAw0O3fuLHP6zp07TY0aNSoxUUm+fgA1xhmFyRhjIiIizN69e93D9evX9xjevn27CQ4Orvxg/z+Xy2Xy8vKsbf9iOSGnLxamanY+aq1yxMfHa9WqVerdu7fS0tLUs2dP+fv7a/bs2bajlZCTk6OMjAzNnTtX+fn56tevnwoLC5WZmak2bdrYjqf/+7//0yOPPKJRo0apefPmtuOUKTc3V9WrVy9zerVq1dxfl21LWlqaRowYoeDg4BLTQkJC9NBDD2nGjBm6+eabLaT7ScOGDbVlyxY1a9as1OlfffWVIiMjKzmVp127dql9+/bu4Ro1asjP76cPlu7SpYtGjx5tI5rb22+/rT/84Q8e4xYsWKAmTZpIkv7yl79o5syZevzxx23Eczt69KgKCwvdwz//PSkuLvaYbkOLFi1+8QtEjxw5UklpyuaEnL72RaxVuow45QDqhNK0evVqzZ07V507d1br1q31wAMP+OQ3zDrhAPrll19q6tSpZU6/4447NG3atEpMVLo777xT48ePV8+ePVWjRg2PaadOndLEiRN11113WUp3jhMOoE4oTJJ09dVXa8uWLWrZsmWp07/66itdffXVlZzK06RJkxQSEmI1w8VwQk5fK0xVuow45QDqhNJ0ww036IYbblBaWpoWLlyoefPmKTk5WcXFxVq5cqWio6NVp04d2zEdcQB1wtkbSRo3bpyWLFmiFi1aKCkpyX2Q2rZtm2bOnKmioiI988wzVjM64QDqhMIknfvdmTBhgnr37l3q786kSZPUu3dvS+nOGTBggMLCwqxmuBhOyOlzhcn2daLKcOLECTN37lxz0003merVqxs/Pz+TlpZm8vPzbUczxhizdu1aM3z4cFOnTh3TpUsX8/LLL5tDhw753P0YP7dt2zYzZswYExERYWrUqGHi4+NtRzI5OTkmKirKREdHm6lTp5rMzEyTmZlpXnjhBRMdHW2ioqJMTk6O1YxNmjQx7733XpnT//a3v5nGjRtXXqAL2Ldvn+nVq5fx8/MzLpfLuFwu4+fnZ3r16uUTH872yCOPmDZt2pR6A+jJkydNmzZtzCOPPGIh2U+aNWtmFi9eXOb0hQsXmqZNm1ZiotLl5OSYiIgI06hRI/Piiy+6f3emTp1qoqOjTWRkpNXfHV+86bI0Tsjpi/eMXBFl5L/54gH0PF8vTWU5e/asee+993zmZ+nrB9CkpCTTrl27Mg+g7dq1M7///e8tJCvbkSNHzLp168wXX3xhjhw5YjuOm68fQI1xRmE6b8+ePSYuLq7E705cXJzHjcE2+OIBtDROyOmLhclljDG2z87YUFRUpA8++EDz5s3T0qVLbccpYfv27Zo7d67eeustHT16VD169PDJnL7shx9+0K5du2SMUfPmzVW3bl3bkSSdu0xz3XXXyd/fv8zLHxs3blR4eLjlpM6wd+9ejRo1SitXrtT5P2cul0s9evTQq6++6r5R1Jbc3Fx17NhRAQEBSkpKUosWLSSd+x1/5ZVXdPbsWW3atMmn9veRI0e0a9cuSVKzZs1Ur149y4lQnvz8/JSTk+NTl5Ku2DLiFL5emnBpvv32W40aNUorVqzwOIDGxcVp5syZaty4seWEzuPLB1BfL0yAbZQRwCJfPXuDiuHLhQmwiTICAACs8vvlWQAAACoOZQQAAFhFGQEAAFZRRgA4QkZGhkJDQ23HAFABKCMA3BISEuRyuTRy5MgS00aPHi2Xy6WEhISLXp/L5VJmZqbXOWJiYpSWluYxrn///tqxY4fX6wLg+ygjADxER0fr3Xff1alTp9zjfvzxRy1YsECNGjWylqtmzZo+9SFNAMoPZQSAh+uuu07R0dFasmSJe9ySJUvUqFEjderUyT2utLMXHTt21LPPPuueLkn33HOPXC6Xe3j37t26++67FR4erqCgIP3qV7/S3//+d/c6unXrpm+//VaPP/64XC6X+5tFS7tMM2vWLDVt2lQBAQFq2bKl3nrrLY/pLpdLb7zxhu655x7VqlVLzZs358MDAR9EGQFQwrBhw5Senu4enjdvnhITE71axz//+U9JUnp6ug4ePOgePnHihO68805lZWVp06ZN6tmzp+Lj47V//35J54rP1VdfrcmTJ+vgwYM6ePBgqet/77339Oijj+qJJ57Qli1b9NBDDykxMVGffPKJx3yTJk1Sv3799NVXX+nOO+/U4MGDK/Wr0QH8MsoIgBLuv/9+rV69Wt9++62+/fZbrVmzRvfff79X62jQoIEkKTQ0VBEREe7hDh066KGHHlK7du3UvHlzPffcc2ratKn7jEW9evXk7++vOnXqKCIiQhEREaWuf9q0aUpISNDDDz+sFi1aKDk5WX379tW0adM85ktISNDAgQPVrFkzTZkyRSdOnNC6deu8/ZEAqECUEQAlNGjQQL1791ZGRobS09PVu3dv1a9fv1zWfeLECT355JNq3bq1QkNDFRQUpK1bt7rPjFysrVu36qabbvIYd9NNN2nr1q0e46699lr3v2vXrq3g4GDl5eVd+hMAUO6q2Q4AwDcNGzZMSUlJkqSZM2eWmO7n56eff5vEmTNnfnG9Tz75pFauXKlp06apWbNmqlmzpn7729/q9OnT5RP8Z6pXr+4x7HK5VFxcXCHbAnBpODMCoFQ9e/bU6dOndebMGcXFxZWY3qBBA4/7OfLz87V3716PeapXr66ioiKPcWvWrFFCQoLuuecetW/fXhEREdq3b5/HPAEBASWW+7nWrVtrzZo1Jdbdpk2bi3l6AHwIZ0YAlMrf3999ycPf37/E9Ntuu00ZGRmKj49XaGioJkyYUGK+mJgYZWVl6aabblJgYKDq1q2r5s2ba8mSJYqPj5fL5dL48eNLnKmIiYnRqlWrNGDAAAUGBpZ6iWjMmDHq16+fOnXqpO7du+uDDz7QkiVLPN6ZA8AZODMCoEzBwcEKDg4udVpKSop+/etf66677lLv3r3Vp08fNW3a1GOe6dOna+XKlYqOjna/LXjGjBmqW7eubrzxRsXHxysuLk7XXXedx3KTJ0/Wvn371LRpU/eNrz/Xp08fvfTSS5o2bZratm2r1157Tenp6erWrdvlP3EAlcplfn7RFwAAoBJxZgQAAFhFGQEAAFZRRgAAgFWUEQAAYBVlBAAAWEUZAQAAVlFGAACAVZQRAABgFWUEAABYRRkBAABWUUYAAIBVlBEAAGDV/wc3GGG98dazGQAAAABJRU5ErkJggg==",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "x = Chr_1_VC_mut_table['Mutation']\n",
    "Chr_1_VC_mut_table['Frequency'] = Chr_1_VC_mut_table['Mut_count']/ Chr_1_VC_mut_table['Nucleotide_count']/Chr_1_VC_mut_table['Total generations']\n",
    "Chr_1_VC_mut_table.plot.bar(x='Mutation', y='Frequency')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "id": "2ae0095b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "186598\n",
      "192625\n",
      "193615\n",
      "173919\n"
     ]
    }
   ],
   "source": [
    "print(VC_chr1_1_4_A)\n",
    "print(VC_chr1_1_4_G)\n",
    "print(VC_chr1_1_4_T)\n",
    "print(VC_chr1_1_4_C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "id": "08de31c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base            Cell_line  A_count  G_count  T_count  \\\n",
       "0       A>C            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "1       A>G            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "2       A>T            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "3       C>A            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "4       C>G            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "5       C>T            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "6       G>A            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "7       G>C            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "8       G>T            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "9       T>A            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "10      T>C            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "11      T>G            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "\n",
       "    C_count  Nucleotide_count  Mut_count  Total generations  \n",
       "0    173919                 0          0              27588  \n",
       "1    173919                 0          0              27588  \n",
       "2    173919                 0          0              27588  \n",
       "3    173919                 0          0              27588  \n",
       "4    173919                 0          0              27588  \n",
       "5    173919                 0          0              27588  \n",
       "6    173919                 0          0              27588  \n",
       "7    173919                 0          0              27588  \n",
       "8    173919                 0          0              27588  \n",
       "9    173919                 0          0              27588  \n",
       "10   173919                 0          0              27588  \n",
       "11   173919                 0          0              27588  "
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_1_4_table = pd.DataFrame(\n",
    "    {\n",
    "        'Mutation': ['A>C', 'A>G', 'A>T', 'C>A', 'C>G', 'C>T', 'G>A', 'G>C', 'G>T', 'T>A', 'T>C', 'T>G'],\n",
    "        'Mutated_base': ['A', 'A', 'A', 'C', 'C', 'C', 'G', 'G', 'G', 'T', 'T' ,'T'],\n",
    "        'Cell_line': ['V. cholerae mut 1/4', 'V. cholerae mut 1/4', 'V. cholerae mut 1/4','V. cholerae mut 1/4', 'V. cholerae mut 1/4', 'V. cholerae mut 1/4', 'V. cholerae mut 1/4', 'V. cholerae mut 1/4', 'V. cholerae mut 1/4', 'V. cholerae mut 1/4', 'V. cholerae mut 1/4', 'V. cholerae mut 1/4'],\n",
    "        'A_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'G_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'T_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'C_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Nucleotide_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Mut_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Total generations' :[27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588]\n",
    "    }\n",
    ")\n",
    "Chr_1_VC_mut_1_4_table['A_count'] = VC_chr1_1_4_A \n",
    "Chr_1_VC_mut_1_4_table['G_count'] = VC_chr1_1_4_G\n",
    "Chr_1_VC_mut_1_4_table['T_count'] = VC_chr1_1_4_T\n",
    "Chr_1_VC_mut_1_4_table['C_count'] = VC_chr1_1_4_C\n",
    "Chr_1_VC_mut_1_4_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "12b9b95f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Nuc_count_1_4(n):\n",
    "    if n == 'A':\n",
    "        return VC_chr1_1_4_A\n",
    "    elif n == 'C':\n",
    "        return VC_chr1_1_4_C\n",
    "    elif n == 'G':\n",
    "        return VC_chr1_1_4_G\n",
    "    else:\n",
    "        return VC_chr1_1_4_T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21170976",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Nuc_count_2_4(n):\n",
    "    if n == 'A':\n",
    "        return VC_chr1_2_4_A\n",
    "    elif n == 'C':\n",
    "        return VC_chr1_2_4_C\n",
    "    elif n == 'G':\n",
    "        return VC_chr1_2_4_G\n",
    "    else:\n",
    "        return VC_chr1_2_4_T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ac79a60",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Nuc_count_3_4(n):\n",
    "    if n == 'A':\n",
    "        return VC_chr1_3_4_A\n",
    "    elif n == 'C':\n",
    "        return VC_chr1_3_4_C\n",
    "    elif n == 'G':\n",
    "        return VC_chr1_3_4_G\n",
    "    else:\n",
    "        return VC_chr1_3_4_T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "619bc4bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Nuc_count_4_4(n):\n",
    "    if n == 'A':\n",
    "        return VC_chr1_4_4_A\n",
    "    elif n == 'C':\n",
    "        return VC_chr1_4_4_C\n",
    "    elif n == 'G':\n",
    "        return VC_chr1_4_4_G\n",
    "    else:\n",
    "        return VC_chr1_4_4_T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "id": "224c8702",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Mut_count_1_4(n):\n",
    "    if n == 'A>G':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['A>G']\n",
    "    elif n == 'A>T':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['A>T']\n",
    "    elif n == 'C>A':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['C>A']\n",
    "    elif n == 'C>G':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['C>G']\n",
    "    elif n == 'C>T':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['C>T']\n",
    "    elif n == 'G>A':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['G>A']\n",
    "    elif n == 'G>C':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['G>C']\n",
    "    elif n == 'G>T':    \n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['G>T']\n",
    "    elif n == 'T>A':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['T>A']\n",
    "    elif n == 'T>C':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['T>C']\n",
    "    elif n == 'T>G':\n",
    "        return Chr_1_VC_mut_1_4['Mutation2'].value_counts()['T>G']\n",
    "    else:\n",
    "        return 0 \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "9797fb01",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Mut_count_2_4(n):\n",
    "    if n == 'A>C':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['A>C']\n",
    "    elif n == 'A>G':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['A>G']\n",
    "    elif n == 'A>T':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['A>T']\n",
    "    elif n == 'C>A':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['C>A']\n",
    "    elif n == 'C>G':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['C>G']\n",
    "    elif n == 'C>T':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['C>T']\n",
    "    elif n == 'G>A':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['G>A']\n",
    "    elif n == 'G>C':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['G>C']\n",
    "    elif n == 'G>T':    \n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['G>T']\n",
    "    elif n == 'T>A':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['T>A']\n",
    "    elif n == 'T>C':\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['T>C']\n",
    "    else:\n",
    "        return Chr_1_VC_mut_2_4['Mutation2'].value_counts()['T>G']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "3bbef267",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Mut_count_3_4(n):\n",
    "    if n == 'A>C':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['A>C']\n",
    "    elif n == 'A>G':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['A>G']\n",
    "    elif n == 'A>T':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['A>T']\n",
    "    elif n == 'C>A':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['C>A']\n",
    "    elif n == 'C>G':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['C>G']\n",
    "    elif n == 'C>T':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['C>T']\n",
    "    elif n == 'G>A':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['G>A']\n",
    "    elif n == 'G>C':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['G>C']\n",
    "    elif n == 'G>T':    \n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['G>T']\n",
    "    elif n == 'T>A':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['T>A']\n",
    "    elif n == 'T>C':\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['T>C']\n",
    "    else:\n",
    "        return Chr_1_VC_mut_3_4['Mutation2'].value_counts()['T>G']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "6397e34e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Mut_count_4_4(n):\n",
    "    if n == 'A>C':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['A>C']\n",
    "    elif n == 'A>G':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['A>G']\n",
    "    elif n == 'A>T':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['A>T']\n",
    "    elif n == 'C>A':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['C>A']\n",
    "    elif n == 'C>G':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['C>G']\n",
    "    elif n == 'C>T':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['C>T']\n",
    "    elif n == 'G>A':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['G>A']\n",
    "    elif n == 'G>C':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['G>C']\n",
    "    elif n == 'G>T':    \n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['G>T']\n",
    "    elif n == 'T>A':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['T>A']\n",
    "    elif n == 'T>C':\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['T>C']\n",
    "    else:\n",
    "        return Chr_1_VC_mut_4_4['Mutation2'].value_counts()['T>G']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "379890d5",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>186598</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>186598</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>186598</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>192625</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>192625</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>192625</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>193615</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>193615</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>193615</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base            Cell_line  A_count  G_count  T_count  \\\n",
       "0       A>C            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "1       A>G            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "2       A>T            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "3       C>A            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "4       C>G            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "5       C>T            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "6       G>A            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "7       G>C            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "8       G>T            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "9       T>A            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "10      T>C            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "11      T>G            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "\n",
       "    C_count  Nucleotide_count  Mut_count  Total generations  \n",
       "0    173919            186598          0              27588  \n",
       "1    173919            186598          0              27588  \n",
       "2    173919            186598          0              27588  \n",
       "3    173919            173919          0              27588  \n",
       "4    173919            173919          0              27588  \n",
       "5    173919            173919          0              27588  \n",
       "6    173919            192625          0              27588  \n",
       "7    173919            192625          0              27588  \n",
       "8    173919            192625          0              27588  \n",
       "9    173919            193615          0              27588  \n",
       "10   173919            193615          0              27588  \n",
       "11   173919            193615          0              27588  "
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_1_4_table['Nucleotide_count'] = Chr_1_VC_mut_1_4_table['Mutated_base'].map(Nuc_count_1_4)\n",
    "Chr_1_VC_mut_1_4_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "66fc1ba0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>186598</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>186598</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>186598</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>173919</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>192625</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>192625</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>192625</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>193615</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>193615</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut 1/4</td>\n",
       "      <td>186598</td>\n",
       "      <td>192625</td>\n",
       "      <td>193615</td>\n",
       "      <td>173919</td>\n",
       "      <td>193615</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base            Cell_line  A_count  G_count  T_count  \\\n",
       "0       A>C            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "1       A>G            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "2       A>T            A  V. cholerae mut 1/4   186598   192625   193615   \n",
       "3       C>A            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "4       C>G            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "5       C>T            C  V. cholerae mut 1/4   186598   192625   193615   \n",
       "6       G>A            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "7       G>C            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "8       G>T            G  V. cholerae mut 1/4   186598   192625   193615   \n",
       "9       T>A            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "10      T>C            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "11      T>G            T  V. cholerae mut 1/4   186598   192625   193615   \n",
       "\n",
       "    C_count  Nucleotide_count  Mut_count  Total generations  \n",
       "0    173919            186598          0              27588  \n",
       "1    173919            186598          0              27588  \n",
       "2    173919            186598          0              27588  \n",
       "3    173919            173919          0              27588  \n",
       "4    173919            173919          0              27588  \n",
       "5    173919            173919          0              27588  \n",
       "6    173919            192625          0              27588  \n",
       "7    173919            192625          0              27588  \n",
       "8    173919            192625          0              27588  \n",
       "9    173919            193615          0              27588  \n",
       "10   173919            193615          0              27588  \n",
       "11   173919            193615          0              27588  "
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_1_4_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "id": "c589d0ed",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'A>T'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\core\\indexes\\base.py:3802\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3801\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m-> 3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\_libs\\index.pyx:138\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\_libs\\index.pyx:165\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5745\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "File \u001b[1;32mpandas\\_libs\\hashtable_class_helper.pxi:5753\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'A>T'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[249], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m Chr_1_VC_mut_1_4_table[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMut_count\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mChr_1_VC_mut_1_4_table\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mMutation\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmap\u001b[49m\u001b[43m(\u001b[49m\u001b[43mMut_count_1_4\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m      2\u001b[0m Chr_1_VC_mut_1_4_table\n\u001b[0;32m      3\u001b[0m Chr_1_VC_mut_table[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMut_count\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m Chr_1_VC_mut_table[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMutation\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mmap(Mut_count)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\core\\series.py:4539\u001b[0m, in \u001b[0;36mSeries.map\u001b[1;34m(self, arg, na_action)\u001b[0m\n\u001b[0;32m   4460\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mmap\u001b[39m(\n\u001b[0;32m   4461\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m   4462\u001b[0m     arg: Callable \u001b[38;5;241m|\u001b[39m Mapping \u001b[38;5;241m|\u001b[39m Series,\n\u001b[0;32m   4463\u001b[0m     na_action: Literal[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[0;32m   4464\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Series:\n\u001b[0;32m   4465\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m   4466\u001b[0m \u001b[38;5;124;03m    Map values of Series according to an input mapping or function.\u001b[39;00m\n\u001b[0;32m   4467\u001b[0m \n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   4537\u001b[0m \u001b[38;5;124;03m    dtype: object\u001b[39;00m\n\u001b[0;32m   4538\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m-> 4539\u001b[0m     new_values \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_map_values\u001b[49m\u001b[43m(\u001b[49m\u001b[43marg\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mna_action\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mna_action\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   4540\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_constructor(new_values, index\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex)\u001b[38;5;241m.\u001b[39m__finalize__(\n\u001b[0;32m   4541\u001b[0m         \u001b[38;5;28mself\u001b[39m, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmap\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   4542\u001b[0m     )\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\core\\base.py:890\u001b[0m, in \u001b[0;36mIndexOpsMixin._map_values\u001b[1;34m(self, mapper, na_action)\u001b[0m\n\u001b[0;32m    887\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(msg)\n\u001b[0;32m    889\u001b[0m \u001b[38;5;66;03m# mapper is a function\u001b[39;00m\n\u001b[1;32m--> 890\u001b[0m new_values \u001b[38;5;241m=\u001b[39m \u001b[43mmap_f\u001b[49m\u001b[43m(\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmapper\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    892\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m new_values\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\_libs\\lib.pyx:2924\u001b[0m, in \u001b[0;36mpandas._libs.lib.map_infer\u001b[1;34m()\u001b[0m\n",
      "Cell \u001b[1;32mIn[248], line 5\u001b[0m, in \u001b[0;36mMut_count_1_4\u001b[1;34m(n)\u001b[0m\n\u001b[0;32m      3\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m Chr_1_VC_mut_1_4[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMutation2\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mvalue_counts()[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mA>G\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m      4\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m n \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mA>T\u001b[39m\u001b[38;5;124m'\u001b[39m:\n\u001b[1;32m----> 5\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mChr_1_VC_mut_1_4\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mMutation2\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mvalue_counts\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mA>T\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\n\u001b[0;32m      6\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m n \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mC>A\u001b[39m\u001b[38;5;124m'\u001b[39m:\n\u001b[0;32m      7\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m Chr_1_VC_mut_1_4[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMutation2\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mvalue_counts()[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mC>A\u001b[39m\u001b[38;5;124m'\u001b[39m]\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\core\\series.py:981\u001b[0m, in \u001b[0;36mSeries.__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m    978\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values[key]\n\u001b[0;32m    980\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m key_is_scalar:\n\u001b[1;32m--> 981\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_value\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    983\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_hashable(key):\n\u001b[0;32m    984\u001b[0m     \u001b[38;5;66;03m# Otherwise index.get_value will raise InvalidIndexError\u001b[39;00m\n\u001b[0;32m    985\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m    986\u001b[0m         \u001b[38;5;66;03m# For labels that don't resolve as scalars like tuples and frozensets\u001b[39;00m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\core\\series.py:1089\u001b[0m, in \u001b[0;36mSeries._get_value\u001b[1;34m(self, label, takeable)\u001b[0m\n\u001b[0;32m   1086\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_values[label]\n\u001b[0;32m   1088\u001b[0m \u001b[38;5;66;03m# Similar to Index.get_value, but we do not fall back to positional\u001b[39;00m\n\u001b[1;32m-> 1089\u001b[0m loc \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlabel\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1090\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mindex\u001b[38;5;241m.\u001b[39m_get_values_for_loc(\u001b[38;5;28mself\u001b[39m, loc, label)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\pandas\\core\\indexes\\base.py:3804\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[1;34m(self, key, method, tolerance)\u001b[0m\n\u001b[0;32m   3802\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[0;32m   3803\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m-> 3804\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[0;32m   3805\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[0;32m   3806\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[0;32m   3807\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[0;32m   3808\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[0;32m   3809\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[1;31mKeyError\u001b[0m: 'A>T'"
     ]
    }
   ],
   "source": [
    "Chr_1_VC_mut_1_4_table['Mut_count'] = Chr_1_VC_mut_1_4_table['Mutation'].map(Mut_count_1_4)\n",
    "Chr_1_VC_mut_1_4_table\n",
    "Chr_1_VC_mut_table['Mut_count'] = Chr_1_VC_mut_table['Mutation'].map(Mut_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "aeee3a39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "C>T    299\n",
       "A>G    256\n",
       "T>C    113\n",
       "G>A     94\n",
       "G>C      4\n",
       "C>A      2\n",
       "A>T      2\n",
       "G>T      2\n",
       "A>C      2\n",
       "T>G      1\n",
       "C>G      1\n",
       "T>A      1\n",
       "Name: Mutation2, dtype: int64"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_1_4_mut['Mutation2'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "2832ba67",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "C>T    299\n",
       "A>G    256\n",
       "T>C    113\n",
       "G>A     94\n",
       "G>C      4\n",
       "C>A      2\n",
       "A>T      2\n",
       "G>T      2\n",
       "A>C      2\n",
       "T>G      1\n",
       "C>G      1\n",
       "T>A      1\n",
       "Name: Mutation2, dtype: int64"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut['Mutation2'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0bab6e19",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
