{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bf8a7126",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import seaborn as sns\n",
    "import Bio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eab066b0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9a48f9c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'C:\\\\Users\\\\antiv\\\\Documents\\\\Python Scripts'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.chdir('C:/Users/antiv/Documents/Python Scripts')\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8b7d0036",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VF_chr1_fasta:\n",
      "NC_006840.2\n",
      "Seq('AAGATCACTTAATATATATAAGATCTTTTAAAGAGATCTTTTATTAGATCTATT...AGA')\n",
      "Lenght of chromosome 1:  2897536\n",
      "number of A: 882812\n",
      "number of T: 886048\n",
      "number of G: 563483\n",
      "number of C: 565193\n",
      "VF_chr2_fasta:\n",
      "NC_006841.2\n",
      "Seq('ACTATGAAAACAATTGTACAGCATAAAAGCAATAATTCAACAAATAGCTGAACA...TTA')\n",
      "Lenght of chromosome 2:  1330333\n",
      "number of A: 423081\n",
      "number of T: 414726\n",
      "number of G: 245073\n",
      "number of C: 247453\n",
      "VC_chr1_fasta:\n",
      "NZ_CP016324.1\n",
      "Seq('GGCAAAATGTGTGAGGATCTTAGTTATCGGTCGAAAAATAATGTGAATAACTTA...ACG')\n",
      "Lenght of chromosome 1:  2987030\n",
      "number of A: 781763\n",
      "number of T: 776269\n",
      "number of G: 714454\n",
      "number of C: 714544\n",
      "VC_chr2_fasta:\n",
      "CP016325.1\n",
      "Seq('GATGAAAAACCGTCTATCCTAACAAGTCTCAGTCAAAACAAGATAAACAGAACA...AGT')\n",
      "Lenght of chromosome 2:  1101931\n",
      "number of A: 291986\n",
      "number of T: 293941\n",
      "number of G: 260201\n",
      "number of C: 255803\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "import os\n",
    "VF_chr1_fasta = SeqIO.read(\"VF_chr_1.fasta\", \"fasta\")\n",
    "print(\"VF_chr1_fasta:\")\n",
    "print(VF_chr1_fasta.id)\n",
    "print(repr(VF_chr1_fasta.seq))\n",
    "print('Lenght of chromosome 1: ', len(VF_chr1_fasta))\n",
    "VF_chr1_A_count = VF_chr1_fasta.count(\"A\")\n",
    "print('number of A:', VF_chr1_A_count)\n",
    "VF_chr1_T_count = VF_chr1_fasta.count(\"T\")\n",
    "print('number of T:', VF_chr1_T_count)\n",
    "VF_chr1_G_count = VF_chr1_fasta.count(\"G\")\n",
    "print('number of G:',VF_chr1_G_count)\n",
    "VF_chr1_C_count = VF_chr1_fasta.count(\"C\")\n",
    "print('number of C:', VF_chr1_C_count)\n",
    "VF_chr2_fasta = SeqIO.read(\"VF_chr_2.fasta\", \"fasta\")\n",
    "print(\"VF_chr2_fasta:\")\n",
    "print(VF_chr2_fasta.id)\n",
    "print(repr(VF_chr2_fasta.seq))\n",
    "print('Lenght of chromosome 2: ', len(VF_chr2_fasta))\n",
    "VF_chr2_A_count = VF_chr2_fasta.count(\"A\")\n",
    "print('number of A:', VF_chr2_A_count)\n",
    "VF_chr2_T_count = VF_chr2_fasta.count(\"T\")\n",
    "print('number of T:', VF_chr2_T_count)\n",
    "VF_chr2_G_count = VF_chr2_fasta.count(\"G\")\n",
    "print('number of G:', VF_chr2_G_count)\n",
    "VF_chr2_C_count = VF_chr2_fasta.count(\"C\")\n",
    "print('number of C:', VF_chr2_C_count)\n",
    "\n",
    "VC_chr1_fasta = SeqIO.read(\"VC_chr_1.fasta\", \"fasta\")\n",
    "print(\"VC_chr1_fasta:\")\n",
    "print(VC_chr1_fasta.id)\n",
    "print(repr(VC_chr1_fasta.seq))\n",
    "print('Lenght of chromosome 1: ', len(VC_chr1_fasta))\n",
    "VC_chr1_A_count = VC_chr1_fasta.count(\"A\")\n",
    "print('number of A:', VC_chr1_A_count)\n",
    "VC_chr1_T_count = VC_chr1_fasta.count(\"T\")\n",
    "print('number of T:', VC_chr1_T_count)\n",
    "VC_chr1_G_count = VC_chr1_fasta.count(\"G\")\n",
    "print('number of G:', VC_chr1_G_count)\n",
    "VC_chr1_C_count = VC_chr1_fasta.count(\"C\")\n",
    "print('number of C:', VC_chr1_C_count)\n",
    "\n",
    "VC_chr2_fasta = SeqIO.read(\"VC_chr_2.fasta\", \"fasta\")\n",
    "print(\"VC_chr2_fasta:\")\n",
    "print(VC_chr2_fasta.id)\n",
    "print(repr(VC_chr2_fasta.seq))\n",
    "print('Lenght of chromosome 2: ', len(VC_chr2_fasta))\n",
    "VC_chr2_A_count = VC_chr2_fasta.count(\"A\")\n",
    "print('number of A:', VC_chr2_A_count)\n",
    "VC_chr2_T_count = VC_chr2_fasta.count(\"T\")\n",
    "print('number of T:', VC_chr2_T_count)\n",
    "VC_chr2_G_count = VC_chr2_fasta.count(\"G\")\n",
    "print('number of G:', VC_chr2_G_count)\n",
    "VC_chr2_C_count = VC_chr2_fasta.count(\"C\")\n",
    "print('number of C:', VC_chr2_C_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1352be64",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Seq('AAGATCACTTAATATATATAAGATCTTTTAAAGAGATCTTTTATTAGATCTATT...TTT')\n",
      "VF_chr1_left_p, A 438645\n",
      "VF_chr1_left_p, G 302214\n",
      "VF_chr1_left_p, T 448583\n",
      "VF_chr1_left_p, C 259326\n",
      "Seq('TCAGAATGATTTTTCACAGACAATACCTAATTATGAAATAAAACGGTTGATACC...AGA')\n",
      "VF_chr1_right_p, T 444167\n",
      "VF_chr1_right_p, C 261269\n",
      "VF_chr1_right_p, A 437465\n",
      "VF_chr1_right_p, G 305867\n",
      "VF_chr1_count_A 876110\n",
      "VF_chr1_count_G 608081\n",
      "VF_chr1_count_T 892750\n",
      "VF_chr1_count_C 520595\n"
     ]
    }
   ],
   "source": [
    "VF_chr1_left_p = VF_chr1_fasta[:1448768]\n",
    "print(repr(VF_chr1_left_p.seq))\n",
    "VF_chr1_left_p_A = VF_chr1_left_p.count(\"A\")\n",
    "print('VF_chr1_left_p, A', VF_chr1_left_p_A)\n",
    "VF_chr1_left_p_G = VF_chr1_left_p.count(\"G\")\n",
    "print('VF_chr1_left_p, G', VF_chr1_left_p_G)\n",
    "VF_chr1_left_p_T = VF_chr1_left_p.count(\"T\")\n",
    "print('VF_chr1_left_p, T', VF_chr1_left_p_T)\n",
    "VF_chr1_left_p_C = VF_chr1_left_p.count(\"C\")\n",
    "print('VF_chr1_left_p, C', VF_chr1_left_p_C)\n",
    "\n",
    "VF_chr1_right_p = VF_chr1_fasta[1448768:]\n",
    "print(repr(VF_chr1_right_p.seq))\n",
    "VF_chr1_right_p_T = VF_chr1_right_p.count(\"A\")\n",
    "print('VF_chr1_right_p, T', VF_chr1_right_p_T)\n",
    "VF_chr1_right_p_C = VF_chr1_right_p.count(\"G\")\n",
    "print('VF_chr1_right_p, C', VF_chr1_right_p_C)\n",
    "VF_chr1_right_p_A = VF_chr1_right_p.count(\"T\")\n",
    "print('VF_chr1_right_p, A', VF_chr1_right_p_A)\n",
    "VF_chr1_right_p_G = VF_chr1_right_p.count(\"C\")\n",
    "print('VF_chr1_right_p, G', VF_chr1_right_p_G)\n",
    "\n",
    "VF_chr1_count_A = VF_chr1_left_p_A + VF_chr1_right_p_A\n",
    "print(\"VF_chr1_count_A\", VF_chr1_count_A )\n",
    "VF_chr1_count_G = VF_chr1_left_p_G + VF_chr1_right_p_G\n",
    "print(\"VF_chr1_count_G\", VF_chr1_count_G )\n",
    "VF_chr1_count_T = VF_chr1_left_p_T + VF_chr1_right_p_T\n",
    "print(\"VF_chr1_count_T\", VF_chr1_count_T )\n",
    "VF_chr1_count_C = VF_chr1_left_p_C + VF_chr1_right_p_C\n",
    "print(\"VF_chr1_count_C\", VF_chr1_count_C )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "dd09af49",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Seq('ACTATGAAAACAATTGTACAGCATAAAAGCAATAATTCAACAAATAGCTGAACA...AAT')\n",
      "VF_chr2_left_p, A 208188\n",
      "VF_chr2_left_p, G 133398\n",
      "VF_chr2_left_p, T 209574\n",
      "VF_chr2_left_p, C 114006\n",
      "Seq('ACCTTAAATATTTTCATAATTTGATTATTTAAAGAGCGATTCTGTAGGAAGTAG...TTA')\n",
      "VF_chr2_right_p, T 214893\n",
      "VF_chr2_right_p, C 111675\n",
      "VF_chr2_right_p, A 205152\n",
      "VF_chr2_right_p, G 133447\n",
      "VF_chr2_count_A 413340\n",
      "VF_chr2_count_G 266845\n",
      "VF_chr2_count_T 424467\n",
      "VF_chr2_count_C 225681\n"
     ]
    }
   ],
   "source": [
    "VF_chr2_left_p = VF_chr2_fasta[:665166]\n",
    "print(repr(VF_chr2_left_p.seq))\n",
    "VF_chr2_left_p_A = VF_chr2_left_p.count(\"A\")\n",
    "print('VF_chr2_left_p, A', VF_chr2_left_p_A)\n",
    "VF_chr2_left_p_G = VF_chr2_left_p.count(\"G\")\n",
    "print('VF_chr2_left_p, G', VF_chr2_left_p_G)\n",
    "VF_chr2_left_p_T = VF_chr2_left_p.count(\"T\")\n",
    "print('VF_chr2_left_p, T', VF_chr2_left_p_T)\n",
    "VF_chr2_left_p_C = VF_chr2_left_p.count(\"C\")\n",
    "print('VF_chr2_left_p, C', VF_chr2_left_p_C)\n",
    "\n",
    "VF_chr2_right_p = VF_chr2_fasta[665166:]\n",
    "print(repr(VF_chr2_right_p.seq))\n",
    "VF_chr2_right_p_T = VF_chr2_right_p.count(\"A\")\n",
    "print('VF_chr2_right_p, T', VF_chr2_right_p_T)\n",
    "VF_chr2_right_p_C = VF_chr2_right_p.count(\"G\")\n",
    "print('VF_chr2_right_p, C', VF_chr2_right_p_C)\n",
    "VF_chr2_right_p_A = VF_chr2_right_p.count(\"T\")\n",
    "print('VF_chr2_right_p, A', VF_chr2_right_p_A)\n",
    "VF_chr2_right_p_G = VF_chr2_right_p.count(\"C\")\n",
    "print('VF_chr2_right_p, G', VF_chr2_right_p_G)\n",
    "\n",
    "VF_chr2_count_A = VF_chr2_left_p_A + VF_chr2_right_p_A\n",
    "print(\"VF_chr2_count_A\", VF_chr2_count_A)\n",
    "VF_chr2_count_G = VF_chr2_left_p_G + VF_chr2_right_p_G\n",
    "print(\"VF_chr2_count_G\", VF_chr2_count_G)\n",
    "VF_chr2_count_T = VF_chr2_left_p_T + VF_chr2_right_p_T\n",
    "print(\"VF_chr2_count_T\", VF_chr2_count_T)\n",
    "VF_chr2_count_C = VF_chr2_left_p_C + VF_chr2_right_p_C\n",
    "print(\"VF_chr2_count_C\", VF_chr2_count_C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b2ebaab0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Seq('GGCAAAATGTGTGAGGATCTTAGTTATCGGTCGAAAAATAATGTGAATAACTTA...CGA')\n",
      "VC_chr1_left_p, A 371449\n",
      "VC_chr1_left_p, G 367754\n",
      "VC_chr1_left_p, T 378575\n",
      "VC_chr1_left_p, C 330737\n",
      "Seq('TGGGGTTATGGAAGACGAATTTTACCCAAACCATCAAGCGGTCGATTTTTATCA...ACG')\n",
      "VC_chr1_right_p, T 410314\n",
      "VC_chr1_right_p, C 346700\n",
      "VC_chr1_right_p, A 397694\n",
      "VC_chr1_right_p, G 383807\n",
      "VC_chr1_count_A 769143\n",
      "VC_chr1_count_G 751561\n",
      "VC_chr1_count_T 788889\n",
      "VC_chr1_count_C 677437\n"
     ]
    }
   ],
   "source": [
    "VC_chr1_left_p = VC_chr1_fasta[:1448515]\n",
    "print(repr(VC_chr1_left_p.seq))\n",
    "VC_chr1_left_p_A = VC_chr1_left_p.count(\"A\")\n",
    "print('VC_chr1_left_p, A', VC_chr1_left_p_A)\n",
    "VC_chr1_left_p_G = VC_chr1_left_p.count(\"G\")\n",
    "print('VC_chr1_left_p, G', VC_chr1_left_p_G)\n",
    "VC_chr1_left_p_T = VC_chr1_left_p.count(\"T\")\n",
    "print('VC_chr1_left_p, T', VC_chr1_left_p_T)\n",
    "VC_chr1_left_p_C = VC_chr1_left_p.count(\"C\")\n",
    "print('VC_chr1_left_p, C', VC_chr1_left_p_C)\n",
    "\n",
    "VC_chr1_right_p = VC_chr1_fasta[1448515:]\n",
    "print(repr(VC_chr1_right_p.seq))\n",
    "VC_chr1_right_p_T = VC_chr1_right_p.count(\"A\")\n",
    "print('VC_chr1_right_p, T', VC_chr1_right_p_T)\n",
    "VC_chr1_right_p_C = VC_chr1_right_p.count(\"G\")\n",
    "print('VC_chr1_right_p, C', VC_chr1_right_p_C)\n",
    "VC_chr1_right_p_A = VC_chr1_right_p.count(\"T\")\n",
    "print('VC_chr1_right_p, A', VC_chr1_right_p_A)\n",
    "VC_chr1_right_p_G = VC_chr1_right_p.count(\"C\")\n",
    "print('VC_chr1_right_p, G', VC_chr1_right_p_G)\n",
    "\n",
    "VC_chr1_count_A = VC_chr1_left_p_A + VC_chr1_right_p_A\n",
    "print(\"VC_chr1_count_A\", VC_chr1_count_A )\n",
    "VC_chr1_count_G = VC_chr1_left_p_G + VC_chr1_right_p_G\n",
    "print(\"VC_chr1_count_G\", VC_chr1_count_G )\n",
    "VC_chr1_count_T = VC_chr1_left_p_T + VC_chr1_right_p_T\n",
    "print(\"VC_chr1_count_T\", VC_chr1_count_T )\n",
    "VC_chr1_count_C = VC_chr1_left_p_C + VC_chr1_right_p_C\n",
    "print(\"VC_chr1_count_C\", VC_chr1_count_C )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b341eb64",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Seq('GATGAAAAACCGTCTATCCTAACAAGTCTCAGTCAAAACAAGATAAACAGAACA...ATA')\n",
      "VC_chr2_left_p, A 147673\n",
      "VC_chr2_left_p, G 135582\n",
      "VC_chr2_left_p, T 153369\n",
      "VC_chr2_left_p, C 119341\n",
      "Seq('AACTGATCTCTTTCAACGCGGTAAATGAGCCAAACTGTTTTACCACATGACGAA...AGT')\n",
      "VC_chr2_right_p, T 144313\n",
      "VC_chr2_right_p, C 124619\n",
      "VC_chr2_right_p, A 140572\n",
      "VC_chr2_right_p, G 136462\n",
      "VC_chr2_count_A 288245\n",
      "VC_chr2_count_G 272044\n",
      "VC_chr2_count_T 297682\n",
      "VC_chr2_count_C 243960\n"
     ]
    }
   ],
   "source": [
    "VC_chr2_left_p = VC_chr2_fasta[:555965]\n",
    "print(repr(VC_chr2_left_p.seq))\n",
    "VC_chr2_left_p_A = VC_chr2_left_p.count(\"A\")\n",
    "print('VC_chr2_left_p, A', VC_chr2_left_p_A)\n",
    "VC_chr2_left_p_G = VC_chr2_left_p.count(\"G\")\n",
    "print('VC_chr2_left_p, G', VC_chr2_left_p_G)\n",
    "VC_chr2_left_p_T = VC_chr2_left_p.count(\"T\")\n",
    "print('VC_chr2_left_p, T', VC_chr2_left_p_T)\n",
    "VC_chr2_left_p_C = VC_chr2_left_p.count(\"C\")\n",
    "print('VC_chr2_left_p, C', VC_chr2_left_p_C)\n",
    "\n",
    "VC_chr2_right_p = VC_chr2_fasta[555965:]\n",
    "print(repr(VC_chr2_right_p.seq))\n",
    "VC_chr2_right_p_T = VC_chr2_right_p.count(\"A\")\n",
    "print('VC_chr2_right_p, T', VC_chr2_right_p_T)\n",
    "VC_chr2_right_p_C = VC_chr2_right_p.count(\"G\")\n",
    "print('VC_chr2_right_p, C', VC_chr2_right_p_C)\n",
    "VC_chr2_right_p_A = VC_chr2_right_p.count(\"T\")\n",
    "print('VC_chr2_right_p, A', VC_chr2_right_p_A)\n",
    "VC_chr2_right_p_G = VC_chr2_right_p.count(\"C\")\n",
    "print('VC_chr2_right_p, G', VC_chr2_right_p_G)\n",
    "\n",
    "VC_chr2_count_A = VC_chr2_left_p_A + VC_chr2_right_p_A\n",
    "print(\"VC_chr2_count_A\", VC_chr2_count_A)\n",
    "VC_chr2_count_G = VC_chr2_left_p_G + VC_chr2_right_p_G\n",
    "print(\"VC_chr2_count_G\", VC_chr2_count_G)\n",
    "VC_chr2_count_T = VC_chr2_left_p_T + VC_chr2_right_p_T\n",
    "print(\"VC_chr2_count_T\", VC_chr2_count_T)\n",
    "VC_chr2_count_C = VC_chr2_left_p_C + VC_chr2_right_p_C\n",
    "print(\"VC_chr2_count_C\", VC_chr2_count_C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d5346662",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr = pd.read_csv('Chr.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c8881f7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Lineage</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Site</th>\n",
       "      <th>AncestralBase</th>\n",
       "      <th>MutatedBase</th>\n",
       "      <th>CODorIG</th>\n",
       "      <th>SYNorNONSYN</th>\n",
       "      <th>AACode</th>\n",
       "      <th>AAChange</th>\n",
       "      <th>Location</th>\n",
       "      <th>...</th>\n",
       "      <th>Consensus</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>AncestralBase_2</th>\n",
       "      <th>MutatedBase_2</th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutation2</th>\n",
       "      <th>A_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>C_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>415883</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>CCA&gt;TCA</td>\n",
       "      <td>P&gt;S</td>\n",
       "      <td>415649:416117</td>\n",
       "      <td>...</td>\n",
       "      <td>98.85</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>29</td>\n",
       "      <td>1</td>\n",
       "      <td>1271811</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>IG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "      <td>891597</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>GTG&gt;GGG</td>\n",
       "      <td>V&gt;G</td>\n",
       "      <td>890434:892516</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>39</td>\n",
       "      <td>1</td>\n",
       "      <td>254586</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>CGC&gt;TGC</td>\n",
       "      <td>R&gt;C</td>\n",
       "      <td>254016:255825</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>49</td>\n",
       "      <td>1</td>\n",
       "      <td>540987</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>COD</td>\n",
       "      <td>syn</td>\n",
       "      <td>AGT&gt;AGC</td>\n",
       "      <td>S&gt;S</td>\n",
       "      <td>540663:542025</td>\n",
       "      <td>...</td>\n",
       "      <td>98.67</td>\n",
       "      <td>V. cholerae WT</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>769143</td>\n",
       "      <td>788889</td>\n",
       "      <td>751561</td>\n",
       "      <td>677437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>491568</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>ACA&gt;GCA</td>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>491432:491978</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>30</td>\n",
       "      <td>2</td>\n",
       "      <td>77590</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>COD</td>\n",
       "      <td>nonsyn</td>\n",
       "      <td>GCA&gt;ACA</td>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>76939:78583</td>\n",
       "      <td>...</td>\n",
       "      <td>98.79</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>232326</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>COD</td>\n",
       "      <td>syn</td>\n",
       "      <td>GCT&gt;GCC</td>\n",
       "      <td>A&gt;A</td>\n",
       "      <td>231905:233111</td>\n",
       "      <td>...</td>\n",
       "      <td>97.98</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>47</td>\n",
       "      <td>2</td>\n",
       "      <td>479984</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>COD</td>\n",
       "      <td>syn</td>\n",
       "      <td>CTG&gt;TTG</td>\n",
       "      <td>L&gt;L</td>\n",
       "      <td>479737:481600</td>\n",
       "      <td>...</td>\n",
       "      <td>100.00</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>22</td>\n",
       "      <td>2</td>\n",
       "      <td>187312</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>COD</td>\n",
       "      <td>syn</td>\n",
       "      <td>CAA&gt;CAG</td>\n",
       "      <td>Q&gt;Q</td>\n",
       "      <td>186771:188700</td>\n",
       "      <td>...</td>\n",
       "      <td>94.87</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>288245</td>\n",
       "      <td>297682</td>\n",
       "      <td>272044</td>\n",
       "      <td>243960</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Lineage  Chromosome     Site AncestralBase MutatedBase CODorIG  \\\n",
       "0          4           1   415883             G           A     COD   \n",
       "1         29           1  1271811             T           C      IG   \n",
       "2         53           1   891597             T           G     COD   \n",
       "3         39           1   254586             G           A     COD   \n",
       "4         49           1   540987             T           C     COD   \n",
       "..       ...         ...      ...           ...         ...     ...   \n",
       "995       24           2   491568             A           G     COD   \n",
       "996       30           2    77590             C           T     COD   \n",
       "997       33           2   232326             A           G     COD   \n",
       "998       47           2   479984             C           T     COD   \n",
       "999       22           2   187312             T           C     COD   \n",
       "\n",
       "    SYNorNONSYN   AACode AAChange       Location  ... Consensus  \\\n",
       "0        nonsyn  CCA>TCA      P>S  415649:416117  ...     98.85   \n",
       "1             -        -        -              -  ...    100.00   \n",
       "2        nonsyn  GTG>GGG      V>G  890434:892516  ...    100.00   \n",
       "3        nonsyn  CGC>TGC      R>C  254016:255825  ...    100.00   \n",
       "4           syn  AGT>AGC      S>S  540663:542025  ...     98.67   \n",
       "..          ...      ...      ...            ...  ...       ...   \n",
       "995      nonsyn  ACA>GCA      T>A  491432:491978  ...    100.00   \n",
       "996      nonsyn  GCA>ACA      A>T    76939:78583  ...     98.79   \n",
       "997         syn  GCT>GCC      A>A  231905:233111  ...     97.98   \n",
       "998         syn  CTG>TTG      L>L  479737:481600  ...    100.00   \n",
       "999         syn  CAA>CAG      Q>Q  186771:188700  ...     94.87   \n",
       "\n",
       "           Cell_line AncestralBase_2  MutatedBase_2  Mutation Mutation2  \\\n",
       "0     V. cholerae WT               G              A       G>A       G>A   \n",
       "1     V. cholerae WT               T              C       T>C       T>C   \n",
       "2     V. cholerae WT               T              G       T>G       T>G   \n",
       "3     V. cholerae WT               G              A       G>A       G>A   \n",
       "4     V. cholerae WT               T              C       T>C       T>C   \n",
       "..               ...             ...            ...       ...       ...   \n",
       "995  V. cholerae mut               A              G       A>G       A>G   \n",
       "996  V. cholerae mut               C              T       C>T       C>T   \n",
       "997  V. cholerae mut               A              G       A>G       A>G   \n",
       "998  V. cholerae mut               C              T       C>T       C>T   \n",
       "999  V. cholerae mut               T              C       T>C       T>C   \n",
       "\n",
       "    A_count T_count G_count C_count  \n",
       "0    769143  788889  751561  677437  \n",
       "1    769143  788889  751561  677437  \n",
       "2    769143  788889  751561  677437  \n",
       "3    769143  788889  751561  677437  \n",
       "4    769143  788889  751561  677437  \n",
       "..      ...     ...     ...     ...  \n",
       "995  288245  297682  272044  243960  \n",
       "996  288245  297682  272044  243960  \n",
       "997  288245  297682  272044  243960  \n",
       "998  288245  297682  272044  243960  \n",
       "999  288245  297682  272044  243960  \n",
       "\n",
       "[1000 rows x 24 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr.head(1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "21e0d159",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_mut = Chr[(Chr[\"Chromosome\"] == 1)  & (Chr[\"Cell_line\"] == 'V. cholerae mut')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "8552a51b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "V. cholerae mut    777\n",
       "Name: Cell_line, dtype: int64"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut[\"Chromosome\"].value_counts()\n",
    "Chr_1_VC_mut[\"Cell_line\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "c9b8d7fa",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_mut_1_4 = Chr_1_VC_mut[Chr_1_VC_mut['Site'] < 746757.5]       \n",
    "Chr_1_VC_mut_2_4 = Chr_1_VC_mut[(Chr_1_VC_mut['Site'] > 746757.5) & (Chr_1_VC_mut['Site'] < 1493515)]\n",
    "Chr_1_VC_mut_3_4 = Chr_1_VC_mut[(Chr_1_VC_mut['Site'] > 1493515) & (Chr_1_VC_mut['Site'] < 2240272.5)]\n",
    "Chr_1_VC_mut_4_4 = Chr_1_VC_mut[Chr_1_VC_mut['Site'] > 2240272.5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "eaa7174d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method NDFrame._add_numeric_operations.<locals>.max of 482    2379167\n",
       "487    2379202\n",
       "490    2321140\n",
       "495    2679004\n",
       "499    2574258\n",
       "        ...   \n",
       "861    2285037\n",
       "863    2248854\n",
       "864    2935412\n",
       "866    2489721\n",
       "873    2931557\n",
       "Name: Site, Length: 208, dtype: int64>"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_4_4.Site.max"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "f9d65f11",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_WT = Chr[(Chr[\"Chromosome\"] == 1)  & (Chr[\"Cell_line\"] == 'V. cholerae WT')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "11ffd769",
   "metadata": {},
   "outputs": [],
   "source": [
    "Chr_1_VC_WT_1_4 = Chr_1_VC_WT[Chr_1_VC_WT['Site'] < 746757.5]       \n",
    "Chr_1_VC_WT_2_4 = Chr_1_VC_WT[(Chr_1_VC_WT['Site'] > 746757.5) & (Chr_1_VC_WT['Site'] < 1493515)]\n",
    "Chr_1_VC_WT_3_4 = Chr_1_VC_WT[(Chr_1_VC_WT['Site'] > 1493515) & (Chr_1_VC_WT['Site'] < 2240272.5)]\n",
    "Chr_1_VC_WT_4_4 = Chr_1_VC_WT[Chr_1_VC_WT['Site'] > 2240272.5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "171ab736",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base        Cell_line  A_count  G_count  T_count  C_count  \\\n",
       "0       A>C            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "1       A>G            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "2       A>T            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "3       C>A            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "4       C>G            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "5       C>T            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "6       G>A            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "7       G>C            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "8       G>T            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "9       T>A            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "10      T>C            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "11      T>G            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "\n",
       "    Nucleotide_count  Mut_count  Total generations  \n",
       "0                  0          0              27588  \n",
       "1                  0          0              27588  \n",
       "2                  0          0              27588  \n",
       "3                  0          0              27588  \n",
       "4                  0          0              27588  \n",
       "5                  0          0              27588  \n",
       "6                  0          0              27588  \n",
       "7                  0          0              27588  \n",
       "8                  0          0              27588  \n",
       "9                  0          0              27588  \n",
       "10                 0          0              27588  \n",
       "11                 0          0              27588  "
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_table = pd.DataFrame(\n",
    "    {\n",
    "        'Mutation': ['A>C', 'A>G', 'A>T', 'C>A', 'C>G', 'C>T', 'G>A', 'G>C', 'G>T', 'T>A', 'T>C', 'T>G'],\n",
    "        'Mutated_base': ['A', 'A', 'A', 'C', 'C', 'C', 'G', 'G', 'G', 'T', 'T' ,'T'],\n",
    "        'Cell_line': ['V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut','V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut', 'V. cholerae mut'],\n",
    "        'A_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'G_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'T_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'C_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Nucleotide_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Mut_count': [0,0,0,0,0,0,0,0,0,0,0,0],\n",
    "        'Total generations' :[27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588, 27588]\n",
    "    }\n",
    ")\n",
    "Chr_1_VC_mut_table['A_count'] = VF_chr1_A_count\n",
    "Chr_1_VC_mut_table['G_count'] = VF_chr1_G_count\n",
    "Chr_1_VC_mut_table['T_count'] = VF_chr1_T_count\n",
    "Chr_1_VC_mut_table['C_count'] = VF_chr1_C_count\n",
    "Chr_1_VC_mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "9e4d0b77",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>882812</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>882812</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>882812</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>565193</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>563483</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>563483</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>563483</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>886048</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>886048</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>886048</td>\n",
       "      <td>0</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base        Cell_line  A_count  G_count  T_count  C_count  \\\n",
       "0       A>C            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "1       A>G            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "2       A>T            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "3       C>A            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "4       C>G            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "5       C>T            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "6       G>A            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "7       G>C            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "8       G>T            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "9       T>A            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "10      T>C            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "11      T>G            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "\n",
       "    Nucleotide_count  Mut_count  Total generations  \n",
       "0             882812          0              27588  \n",
       "1             882812          0              27588  \n",
       "2             882812          0              27588  \n",
       "3             565193          0              27588  \n",
       "4             565193          0              27588  \n",
       "5             565193          0              27588  \n",
       "6             563483          0              27588  \n",
       "7             563483          0              27588  \n",
       "8             563483          0              27588  \n",
       "9             886048          0              27588  \n",
       "10            886048          0              27588  \n",
       "11            886048          0              27588  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def Nuc_count(n):\n",
    "    if n == 'A':\n",
    "        return VF_chr1_A_count\n",
    "    elif n == 'C':\n",
    "        return VF_chr1_C_count\n",
    "    elif n == 'G':\n",
    "        return VF_chr1_G_count\n",
    "    else:\n",
    "        return VF_chr1_T_count\n",
    "\n",
    "Chr_1_VC_mut_table['Nucleotide_count'] = Chr_1_VC_mut_table['Mutated_base'].map(Nuc_count)\n",
    "Chr_1_VC_mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "7f48e53f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Mut_count(n):\n",
    "    if n == 'A>C':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['A>C']\n",
    "    elif n == 'A>G':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['A>G']\n",
    "    elif n == 'A>T':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['A>T']\n",
    "    elif n == 'C>A':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['C>A']\n",
    "    elif n == 'C>G':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['C>G']\n",
    "    elif n == 'C>T':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['C>T']\n",
    "    elif n == 'G>A':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['G>A']\n",
    "    elif n == 'G>C':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['G>C']\n",
    "    elif n == 'G>T':    \n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['G>T']\n",
    "    elif n == 'T>A':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['T>A']\n",
    "    elif n == 'T>C':\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['T>C']\n",
    "    else:\n",
    "        return Chr_1_VC_mut['Mutation2'].value_counts()['T>G']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "b1d75629",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mutation</th>\n",
       "      <th>Mutated_base</th>\n",
       "      <th>Cell_line</th>\n",
       "      <th>A_count</th>\n",
       "      <th>G_count</th>\n",
       "      <th>T_count</th>\n",
       "      <th>C_count</th>\n",
       "      <th>Nucleotide_count</th>\n",
       "      <th>Mut_count</th>\n",
       "      <th>Total generations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A&gt;C</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>882812</td>\n",
       "      <td>2</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>882812</td>\n",
       "      <td>256</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A&gt;T</td>\n",
       "      <td>A</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>882812</td>\n",
       "      <td>2</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C&gt;A</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>565193</td>\n",
       "      <td>2</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C&gt;G</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>565193</td>\n",
       "      <td>1</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>C</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>565193</td>\n",
       "      <td>299</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>563483</td>\n",
       "      <td>94</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>G&gt;C</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>563483</td>\n",
       "      <td>4</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>G&gt;T</td>\n",
       "      <td>G</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>563483</td>\n",
       "      <td>2</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>886048</td>\n",
       "      <td>1</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T&gt;C</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>886048</td>\n",
       "      <td>113</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>T</td>\n",
       "      <td>V. cholerae mut</td>\n",
       "      <td>882812</td>\n",
       "      <td>563483</td>\n",
       "      <td>886048</td>\n",
       "      <td>565193</td>\n",
       "      <td>886048</td>\n",
       "      <td>1</td>\n",
       "      <td>27588</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Mutation Mutated_base        Cell_line  A_count  G_count  T_count  C_count  \\\n",
       "0       A>C            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "1       A>G            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "2       A>T            A  V. cholerae mut   882812   563483   886048   565193   \n",
       "3       C>A            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "4       C>G            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "5       C>T            C  V. cholerae mut   882812   563483   886048   565193   \n",
       "6       G>A            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "7       G>C            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "8       G>T            G  V. cholerae mut   882812   563483   886048   565193   \n",
       "9       T>A            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "10      T>C            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "11      T>G            T  V. cholerae mut   882812   563483   886048   565193   \n",
       "\n",
       "    Nucleotide_count  Mut_count  Total generations  \n",
       "0             882812          2              27588  \n",
       "1             882812        256              27588  \n",
       "2             882812          2              27588  \n",
       "3             565193          2              27588  \n",
       "4             565193          1              27588  \n",
       "5             565193        299              27588  \n",
       "6             563483         94              27588  \n",
       "7             563483          4              27588  \n",
       "8             563483          2              27588  \n",
       "9             886048          1              27588  \n",
       "10            886048        113              27588  \n",
       "11            886048          1              27588  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chr_1_VC_mut_table['Mut_count'] = Chr_1_VC_mut_table['Mutation'].map(Mut_count)\n",
    "Chr_1_VC_mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "80a5f9ef",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "`bins` must increase monotonically, when an array",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[95], line 5\u001b[0m\n\u001b[0;32m      3\u001b[0m x \u001b[38;5;241m=\u001b[39m Chr_1_VC_mut_table[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMutation\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m      4\u001b[0m y \u001b[38;5;241m=\u001b[39m Chr_1_VC_mut_table[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNucleotide_count\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m/\u001b[39m Chr_1_VC_mut_table[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mMut_count\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m/\u001b[39mChr_1_VC_mut_table[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mTotal generations\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[1;32m----> 5\u001b[0m \u001b[43mplt\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhist\u001b[49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[43m,\u001b[49m\u001b[43my\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m      6\u001b[0m plt\u001b[38;5;241m.\u001b[39mshow()\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\matplotlib\\pyplot.py:2618\u001b[0m, in \u001b[0;36mhist\u001b[1;34m(x, bins, range, density, weights, cumulative, bottom, histtype, align, orientation, rwidth, log, color, label, stacked, data, **kwargs)\u001b[0m\n\u001b[0;32m   2612\u001b[0m \u001b[38;5;129m@_copy_docstring_and_deprecators\u001b[39m(Axes\u001b[38;5;241m.\u001b[39mhist)\n\u001b[0;32m   2613\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mhist\u001b[39m(\n\u001b[0;32m   2614\u001b[0m         x, bins\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;28mrange\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, density\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m, weights\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[0;32m   2615\u001b[0m         cumulative\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m, bottom\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, histtype\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbar\u001b[39m\u001b[38;5;124m'\u001b[39m, align\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmid\u001b[39m\u001b[38;5;124m'\u001b[39m,\n\u001b[0;32m   2616\u001b[0m         orientation\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mvertical\u001b[39m\u001b[38;5;124m'\u001b[39m, rwidth\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, log\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m, color\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m,\n\u001b[0;32m   2617\u001b[0m         label\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, stacked\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m, \u001b[38;5;241m*\u001b[39m, data\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m-> 2618\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mgca\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhist\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m   2619\u001b[0m \u001b[43m        \u001b[49m\u001b[43mx\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbins\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbins\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mrange\u001b[39;49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mrange\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdensity\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdensity\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mweights\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mweights\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   2620\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcumulative\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcumulative\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbottom\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mbottom\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhisttype\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mhisttype\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   2621\u001b[0m \u001b[43m        \u001b[49m\u001b[43malign\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43malign\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43morientation\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43morientation\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrwidth\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrwidth\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlog\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mlog\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   2622\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcolor\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcolor\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlabel\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mlabel\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstacked\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mstacked\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   2623\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m{\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mdata\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m}\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mif\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mis\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mnot\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43;01melse\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43m{\u001b[49m\u001b[43m}\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\matplotlib\\__init__.py:1459\u001b[0m, in \u001b[0;36m_preprocess_data.<locals>.inner\u001b[1;34m(ax, data, *args, **kwargs)\u001b[0m\n\u001b[0;32m   1456\u001b[0m \u001b[38;5;129m@functools\u001b[39m\u001b[38;5;241m.\u001b[39mwraps(func)\n\u001b[0;32m   1457\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21minner\u001b[39m(ax, \u001b[38;5;241m*\u001b[39margs, data\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m   1458\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m data \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m-> 1459\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[43max\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;28;43mmap\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43msanitize_sequence\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43margs\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1461\u001b[0m     bound \u001b[38;5;241m=\u001b[39m new_sig\u001b[38;5;241m.\u001b[39mbind(ax, \u001b[38;5;241m*\u001b[39margs, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m   1462\u001b[0m     auto_label \u001b[38;5;241m=\u001b[39m (bound\u001b[38;5;241m.\u001b[39marguments\u001b[38;5;241m.\u001b[39mget(label_namer)\n\u001b[0;32m   1463\u001b[0m                   \u001b[38;5;129;01mor\u001b[39;00m bound\u001b[38;5;241m.\u001b[39mkwargs\u001b[38;5;241m.\u001b[39mget(label_namer))\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\matplotlib\\axes\\_axes.py:6790\u001b[0m, in \u001b[0;36mAxes.hist\u001b[1;34m(self, x, bins, range, density, weights, cumulative, bottom, histtype, align, orientation, rwidth, log, color, label, stacked, **kwargs)\u001b[0m\n\u001b[0;32m   6786\u001b[0m \u001b[38;5;66;03m# Loop through datasets\u001b[39;00m\n\u001b[0;32m   6787\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(nx):\n\u001b[0;32m   6788\u001b[0m     \u001b[38;5;66;03m# this will automatically overwrite bins,\u001b[39;00m\n\u001b[0;32m   6789\u001b[0m     \u001b[38;5;66;03m# so that each histogram uses the same bins\u001b[39;00m\n\u001b[1;32m-> 6790\u001b[0m     m, bins \u001b[38;5;241m=\u001b[39m \u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhistogram\u001b[49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbins\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mweights\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mw\u001b[49m\u001b[43m[\u001b[49m\u001b[43mi\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mhist_kwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   6791\u001b[0m     tops\u001b[38;5;241m.\u001b[39mappend(m)\n\u001b[0;32m   6792\u001b[0m tops \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39marray(tops, \u001b[38;5;28mfloat\u001b[39m)  \u001b[38;5;66;03m# causes problems later if it's an int\u001b[39;00m\n",
      "File \u001b[1;32m<__array_function__ internals>:200\u001b[0m, in \u001b[0;36mhistogram\u001b[1;34m(*args, **kwargs)\u001b[0m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\numpy\\lib\\histograms.py:780\u001b[0m, in \u001b[0;36mhistogram\u001b[1;34m(a, bins, range, density, weights)\u001b[0m\n\u001b[0;32m    680\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m    681\u001b[0m \u001b[38;5;124;03mCompute the histogram of a dataset.\u001b[39;00m\n\u001b[0;32m    682\u001b[0m \n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    776\u001b[0m \n\u001b[0;32m    777\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m    778\u001b[0m a, weights \u001b[38;5;241m=\u001b[39m _ravel_and_check_weights(a, weights)\n\u001b[1;32m--> 780\u001b[0m bin_edges, uniform_bins \u001b[38;5;241m=\u001b[39m \u001b[43m_get_bin_edges\u001b[49m\u001b[43m(\u001b[49m\u001b[43ma\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbins\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mrange\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mweights\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    782\u001b[0m \u001b[38;5;66;03m# Histogram is an integer or a float array depending on the weights.\u001b[39;00m\n\u001b[0;32m    783\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m weights \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[1;32m~\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\numpy\\lib\\histograms.py:431\u001b[0m, in \u001b[0;36m_get_bin_edges\u001b[1;34m(a, bins, range, weights)\u001b[0m\n\u001b[0;32m    429\u001b[0m     bin_edges \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39masarray(bins)\n\u001b[0;32m    430\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m np\u001b[38;5;241m.\u001b[39many(bin_edges[:\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m] \u001b[38;5;241m>\u001b[39m bin_edges[\u001b[38;5;241m1\u001b[39m:]):\n\u001b[1;32m--> 431\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[0;32m    432\u001b[0m             \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m`bins` must increase monotonically, when an array\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m    434\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    435\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m`bins` must be 1d, when an array\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[1;31mValueError\u001b[0m: `bins` must increase monotonically, when an array"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAiMAAAGiCAYAAAA1LsZRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAYY0lEQVR4nO3db2ydZeH/8U/bsVMJtkwmHZsl8y9TgQ03VgsYQ1KdkUxJNClD2DIFAxKCNEYYfzYmfwrqyB5suIhDJREZohLjyAhWQY1NJpsjEhkEATfQdluQlg3tpO3vgfmW9LcNdsbGZcfrlZwHu3pd932dR33vvu+eUzM8PDwcAIBCaktvAAB4axMjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUVXHyG9/+9vMnTs3kydPTk1NTe67777XXfPQQw/lIx/5SCqVSt73vvflBz/4wQFsFQA4HFUdI7t27cr06dOzcuXK/Zr/zDPP5KyzzsqZZ56ZTZs25atf/WouuOCCPPDAA1VvFgA4/NS8kS/Kq6mpyc9//vOcffbZ+5xzxRVXZO3atXnsscdGxs4555y8+OKLWbdu3YGeGgA4TIw71Cfo7u5OW1vbqLE5c+bkq1/96j7XDAwMZGBgYOTfQ0NDeeGFF3LMMcekpqbmUG0VADiIhoeH89JLL2Xy5Mmprd33zZhDHiM9PT1pamoaNdbU1JT+/v7861//ytve9rY91nR2dmbp0qWHemsAwJtg69atede73rXPnx/yGDkQixYtSkdHx8i/+/r6cvzxx2fr1q1paGgouDMAYH/19/enubk5b3/7219z3iGPkUmTJqW3t3fUWG9vbxoaGvZ6VSRJKpVKKpXKHuMNDQ1iBADGmNd7xOKQf85Ia2trurq6Ro09+OCDaW1tPdSnBgDGgKpjZOfOndm0aVM2bdqU5L9/urtp06Zs2bIlyX9vscyfP39k/kUXXZSnn346X//617N58+bcdtttueeee3L55ZcfnHcAAIxpVcfII488klNOOSWnnHJKkqSjoyOnnHJKFi9enCT5xz/+MRImSfLud787a9euzYMPPpjp06dn2bJl+d73vpc5c+YcpLcAAIxlb+hzRt4s/f39aWxsTF9fn2dGAGCM2N/f376bBgAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIo6oBhZuXJlpk6dmvr6+rS0tGT9+vWvOX/58uU54YQT8ra3vS3Nzc25/PLL8+9///uANgwAHF6qjpE1a9ako6MjS5YsycaNGzN9+vTMmTMn27Zt2+v8u+66K1deeWWWLFmSxx9/PKtXr86aNWty1VVXveHNAwBjX9Uxcuutt+bCCy/MwoUL86EPfSirVq3KkUcemTvuuGOv8//whz/k9NNPz7nnnpupU6fmk5/8ZObNm/e6V1MAgLeGqmJk9+7d2bBhQ9ra2l49QG1t2tra0t3dvdc1p512WjZs2DASH08//XTuv//+fPrTn97neQYGBtLf3z/qBQAcnsZVM3nHjh0ZHBxMU1PTqPGmpqZs3rx5r2vOPffc7NixI2eccUaGh4fzyiuv5KKLLnrN2zSdnZ1ZunRpNVsDAMaoQ/7XNA899FBuuumm3Hbbbdm4cWN+9rOfZe3atbn++uv3uWbRokXp6+sbeW3duvVQbxMAKKSqKyMTJ05MXV1dent7R4339vZm0qRJe11z7bXX5vzzz88FF1yQJDnppJOya9eufPnLX87VV1+d2to9e6hSqaRSqVSzNQBgjKrqysj48eMzc+bMdHV1jYwNDQ2lq6srra2te13z8ssv7xEcdXV1SZLh4eFq9wsAHGaqujKSJB0dHVmwYEFmzZqV2bNnZ/ny5dm1a1cWLlyYJJk/f36mTJmSzs7OJMncuXNz66235pRTTklLS0ueeuqpXHvttZk7d+5IlAAAb11Vx0h7e3u2b9+exYsXp6enJzNmzMi6detGHmrdsmXLqCsh11xzTWpqanLNNdfk+eefzzvf+c7MnTs3N95448F7FwDAmFUzPAbulfT396exsTF9fX1paGgovR0AYD/s7+9v300DABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARR1QjKxcuTJTp05NfX19Wlpasn79+tec/+KLL+aSSy7Jcccdl0qlkg984AO5//77D2jDAMDhZVy1C9asWZOOjo6sWrUqLS0tWb58eebMmZMnnngixx577B7zd+/enU984hM59thjc++992bKlCn529/+lqOPPvpg7B8AGONqhoeHh6tZ0NLSklNPPTUrVqxIkgwNDaW5uTmXXnpprrzyyj3mr1q1Kt/61reyefPmHHHEEQe0yf7+/jQ2Nqavry8NDQ0HdAwA4M21v7+/q7pNs3v37mzYsCFtbW2vHqC2Nm1tbenu7t7rml/84hdpbW3NJZdckqamppx44om56aabMjg4uM/zDAwMpL+/f9QLADg8VRUjO3bsyODgYJqamkaNNzU1paenZ69rnn766dx7770ZHBzM/fffn2uvvTbLli3LDTfcsM/zdHZ2prGxceTV3NxczTYBgDHkkP81zdDQUI499th897vfzcyZM9Pe3p6rr746q1at2ueaRYsWpa+vb+S1devWQ71NAKCQqh5gnThxYurq6tLb2ztqvLe3N5MmTdrrmuOOOy5HHHFE6urqRsY++MEPpqenJ7t378748eP3WFOpVFKpVKrZGgAwRlV1ZWT8+PGZOXNmurq6RsaGhobS1dWV1tbWva45/fTT89RTT2VoaGhk7Mknn8xxxx231xABAN5aqr5N09HRkdtvvz0//OEP8/jjj+fiiy/Orl27snDhwiTJ/Pnzs2jRopH5F198cV544YVcdtllefLJJ7N27drcdNNNueSSSw7euwAAxqyqP2ekvb0927dvz+LFi9PT05MZM2Zk3bp1Iw+1btmyJbW1rzZOc3NzHnjggVx++eU5+eSTM2XKlFx22WW54oorDt67AADGrKo/Z6QEnzMCAGPPIfmcEQCAg02MAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKOqAYWblyZaZOnZr6+vq0tLRk/fr1+7Xu7rvvTk1NTc4+++wDOS0AcBiqOkbWrFmTjo6OLFmyJBs3bsz06dMzZ86cbNu27TXXPfvss/na176Wj33sYwe8WQDg8FN1jNx666258MILs3DhwnzoQx/KqlWrcuSRR+aOO+7Y55rBwcF84QtfyNKlS/Oe97zndc8xMDCQ/v7+US8A4PBUVYzs3r07GzZsSFtb26sHqK1NW1tburu797nuG9/4Ro499th86Utf2q/zdHZ2prGxceTV3NxczTYBgDGkqhjZsWNHBgcH09TUNGq8qakpPT09e13z+9//PqtXr87tt9++3+dZtGhR+vr6Rl5bt26tZpsAwBgy7lAe/KWXXsr555+f22+/PRMnTtzvdZVKJZVK5RDuDAD4X1FVjEycODF1dXXp7e0dNd7b25tJkybtMf+vf/1rnn322cydO3dkbGho6L8nHjcuTzzxRN773vceyL4BgMNEVbdpxo8fn5kzZ6arq2tkbGhoKF1dXWltbd1j/rRp0/LnP/85mzZtGnl95jOfyZlnnplNmzZ5FgQAqP42TUdHRxYsWJBZs2Zl9uzZWb58eXbt2pWFCxcmSebPn58pU6aks7Mz9fX1OfHEE0etP/roo5Nkj3EA4K2p6hhpb2/P9u3bs3jx4vT09GTGjBlZt27dyEOtW7ZsSW2tD3YFAPZPzfDw8HDpTbye/v7+NDY2pq+vLw0NDaW3AwDsh/39/e0SBgBQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABR1QDGycuXKTJ06NfX19Wlpacn69ev3Off222/Pxz72sUyYMCETJkxIW1vba84HAN5aqo6RNWvWpKOjI0uWLMnGjRszffr0zJkzJ9u2bdvr/Iceeijz5s3Lb37zm3R3d6e5uTmf/OQn8/zzz7/hzQMAY1/N8PDwcDULWlpacuqpp2bFihVJkqGhoTQ3N+fSSy/NlVde+brrBwcHM2HChKxYsSLz58/f65yBgYEMDAyM/Lu/vz/Nzc3p6+tLQ0NDNdsFAArp7+9PY2Pj6/7+rurKyO7du7Nhw4a0tbW9eoDa2rS1taW7u3u/jvHyyy/nP//5T97xjnfsc05nZ2caGxtHXs3NzdVsEwAYQ6qKkR07dmRwcDBNTU2jxpuamtLT07Nfx7jiiisyefLkUUHz/1u0aFH6+vpGXlu3bq1mmwDAGDLuzTzZzTffnLvvvjsPPfRQ6uvr9zmvUqmkUqm8iTsDAEqpKkYmTpyYurq69Pb2jhrv7e3NpEmTXnPtt7/97dx888351a9+lZNPPrn6nQIAh6WqbtOMHz8+M2fOTFdX18jY0NBQurq60traus913/zmN3P99ddn3bp1mTVr1oHvFgA47FR9m6ajoyMLFizIrFmzMnv27Cxfvjy7du3KwoULkyTz58/PlClT0tnZmSS55ZZbsnjx4tx1112ZOnXqyLMlRx11VI466qiD+FYAgLGo6hhpb2/P9u3bs3jx4vT09GTGjBlZt27dyEOtW7ZsSW3tqxdcvvOd72T37t35/Oc/P+o4S5YsyXXXXffGdg8AjHlVf85ICfv7d8oAwP+OQ/I5IwAAB5sYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUJUYAgKLECABQlBgBAIoSIwBAUWIEAChKjAAARYkRAKAoMQIAFCVGAICixAgAUJQYAQCKEiMAQFFiBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoCgxAgAUdUAxsnLlykydOjX19fVpaWnJ+vXrX3P+T37yk0ybNi319fU56aSTcv/99x/QZgGAw0/VMbJmzZp0dHRkyZIl2bhxY6ZPn545c+Zk27Zte53/hz/8IfPmzcuXvvSl/OlPf8rZZ5+ds88+O4899tgb3jwAMPbVDA8PD1ezoKWlJaeeempWrFiRJBkaGkpzc3MuvfTSXHnllXvMb29vz65du/LLX/5yZOyjH/1oZsyYkVWrVu31HAMDAxkYGBj5d19fX44//vhs3bo1DQ0N1WwXACikv78/zc3NefHFF9PY2LjPeeOqOeju3buzYcOGLFq0aGSstrY2bW1t6e7u3uua7u7udHR0jBqbM2dO7rvvvn2ep7OzM0uXLt1jvLm5uZrtAgD/A1566aWDFyM7duzI4OBgmpqaRo03NTVl8+bNe13T09Oz1/k9PT37PM+iRYtGBczQ0FBeeOGFHHPMMampqalmy8D/uP/7n5Mrn3D4GR4ezksvvZTJkye/5ryqYuTNUqlUUqlURo0dffTRZTYDvCkaGhrECByGXuuKyP+p6gHWiRMnpq6uLr29vaPGe3t7M2nSpL2umTRpUlXzAYC3lqpiZPz48Zk5c2a6urpGxoaGhtLV1ZXW1ta9rmltbR01P0kefPDBfc4HAN5aqr5N09HRkQULFmTWrFmZPXt2li9fnl27dmXhwoVJkvnz52fKlCnp7OxMklx22WX5+Mc/nmXLluWss87K3XffnUceeSTf/e53D+47AcakSqWSJUuW7HFrFnjrqDpG2tvbs3379ixevDg9PT2ZMWNG1q1bN/KQ6pYtW1Jb++oFl9NOOy133XVXrrnmmlx11VV5//vfn/vuuy8nnnjiwXsXwJhVqVRy3XXXld4GUFDVnzMCAHAw+W4aAKAoMQIAFCVGAICixAgAUJQYAQ6q7u7u1NXV5ayzznrDx+rv78/VV1+dadOmpb6+PpMmTUpbW1t+9rOfxbP3cPjw1zTAQXXBBRfkqKOOyurVq/PEE0+M+k6KLVu25Pjjj9+v47z44os544wz0tfXlxtuuCGnnnpqxo0bl4cffji33HJLHnnkEV8TAYcJMQIcNDt37sxxxx2XRx55JEuWLMnJJ5+cq666auTndXV1OfPMM7NgwYJ87nOfy5FHHrnPY33lK1/JnXfemSeffHKPL9nauXNn6uvrM27c/+TXawFVcpsGOGjuueeeTJs2LSeccELOO++83HHHHaNup/zlL3/J7Nmzc80116SpqSlf/OIX8/DDD+9xy2VoaCh33313vvCFL+z12z6POuooIQKHETECHDSrV6/OeeedlyT51Kc+lb6+vjz88MMjPz/hhBNy00035dlnn80vfvGLDA8PZ+7cuXnve9+b6667Ls8880ySZMeOHfnnP/+ZadOmFXkfwJtLjAAHxRNPPJH169dn3rx5SZJx48alvb09q1ev3mNuTU1NzjzzzHz/+9/Pc889l9bW1ixdujSXX355kng4Fd5iXOcEDorVq1fnlVdeGXVbZXh4OJVKJStWrEhjY+Oo+Rs3bsydd96ZH//4x6mpqUlHR0cuuOCCJMk73/nOHH300dm8efOb+h6AMlwZAd6wV155JXfeeWeWLVuWTZs2jbweffTRTJ48OT/+8Y+TJM8991xuvvnmfPjDH85pp52W559/PqtXr85zzz2XZcuW5YMf/GCSpLa2Nuecc05+9KMf5e9///se59u5c2deeeWVN/U9AoeOv6YB3rD77rsv7e3t2bZt2x5XQK644or8+te/zh//+MfU1dVl1qxZWbBgQebNm5cJEybs85gvvPBCTj/99OzcuTM33nhjZs2alSOOOCK/+93v0tnZmT/+8Y/+tBcOE2IEeMPmzp2boaGhrF27do+frV+/Pi0tLXn00Uczfvz4qh5K7evry80335yf/vSn+dvf/pYJEybkpJNOyiWXXJLPfvazqampOZhvAyhEjAAARXlmBAAoSowAAEWJEQCgKDECABQlRgCAosQIAFCUGAEAihIjAEBRYgQAKEqMAABFiREAoKj/B0E6/XmSB18FAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "x = Chr_1_VC_mut_table['Mutation']\n",
    "Chr_1_VC_mut_table['Frequency'] = Chr_1_VC_mut_table['Nucleotide_count']/ Chr_1_VC_mut_table['Mut_count']/Chr_1_VC_mut_table['Total generations']\n",
    "Chr_1_VC_mut_table.plot.hist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08de31c2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
